// oci-identity/lib/model/bulk-edit-operation-details.js
"use strict";
/**
 * Identity and Access Management Service API
 * APIs for managing users, groups, compartments, policies, and identity domains.
 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BulkEditOperationDetails = void 0;
var BulkEditOperationDetails;
(function (BulkEditOperationDetails) {
    let OperationType;
    (function (OperationType) {
        OperationType["AddWhereAbsent"] = "ADD_WHERE_ABSENT";
        OperationType["SetWherePresent"] = "SET_WHERE_PRESENT";
        OperationType["AddOrSet"] = "ADD_OR_SET";
        OperationType["Remove"] = "REMOVE";
    })(OperationType = BulkEditOperationDetails.OperationType || (BulkEditOperationDetails.OperationType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    BulkEditOperationDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    BulkEditOperationDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(BulkEditOperationDetails = exports.BulkEditOperationDetails || (exports.BulkEditOperationDetails = {}));
