// oci-identity/lib/model/iam-work-request-summary.js
"use strict";
/**
 * Identity and Access Management Service API
 * APIs for managing users, groups, compartments, policies, and identity domains.
 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.IamWorkRequestSummary = void 0;
const model = __importStar(require("../model"));
var IamWorkRequestSummary;
(function (IamWorkRequestSummary) {
    let OperationType;
    (function (OperationType) {
        OperationType["CreateDomain"] = "CREATE_DOMAIN";
        OperationType["ReplicateDomainToRegion"] = "REPLICATE_DOMAIN_TO_REGION";
        OperationType["UpdateDomain"] = "UPDATE_DOMAIN";
        OperationType["ActivateDomain"] = "ACTIVATE_DOMAIN";
        OperationType["DeactivateDomain"] = "DEACTIVATE_DOMAIN";
        OperationType["DeleteDomain"] = "DELETE_DOMAIN";
        OperationType["ChangeCompartmentForDomain"] = "CHANGE_COMPARTMENT_FOR_DOMAIN";
        OperationType["ChangeLicenseTypeForDomain"] = "CHANGE_LICENSE_TYPE_FOR_DOMAIN";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        OperationType["UnknownValue"] = "UNKNOWN_VALUE";
    })(OperationType = IamWorkRequestSummary.OperationType || (IamWorkRequestSummary.OperationType = {}));
    let Status;
    (function (Status) {
        Status["Accepted"] = "ACCEPTED";
        Status["InProgress"] = "IN_PROGRESS";
        Status["Failed"] = "FAILED";
        Status["Succeeded"] = "SUCCEEDED";
        Status["Canceling"] = "CANCELING";
        Status["Canceled"] = "CANCELED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = IamWorkRequestSummary.Status || (IamWorkRequestSummary.Status = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "resources": obj.resources
                ? obj.resources.map(item => {
                    return model.IamWorkRequestResource.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    IamWorkRequestSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "resources": obj.resources
                ? obj.resources.map(item => {
                    return model.IamWorkRequestResource.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    IamWorkRequestSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(IamWorkRequestSummary = exports.IamWorkRequestSummary || (exports.IamWorkRequestSummary = {}));
