// oci-identity/lib/model/index.js
"use strict";
/**
 * Identity and Access Management Service API
 * APIs for managing users, groups, compartments, policies, and identity domains.
 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DomainReplicationStates = exports.DomainReplication = exports.Domain = exports.DbCredentialSummary = exports.DbCredential = exports.CustomerSecretKeySummary = exports.CustomerSecretKey = exports.CreateUserDetails = exports.CreateTagNamespaceDetails = exports.CreateTagDetails = exports.CreateTagDefaultDetails = exports.CreateSwiftPasswordDetails = exports.CreateSmtpCredentialDetails = exports.CreateRegionSubscriptionDetails = exports.CreatePolicyDetails = exports.CreateOAuth2ClientCredentialDetails = exports.CreateNetworkSourceDetails = exports.CreateIdpGroupMappingDetails = exports.CreateIdentityProviderDetails = exports.CreateGroupDetails = exports.CreateDynamicGroupDetails = exports.CreateDomainDetails = exports.CreateDbCredentialDetails = exports.CreateCustomerSecretKeyDetails = exports.CreateCompartmentDetails = exports.CreateAuthTokenDetails = exports.CreateApiKeyDetails = exports.Compartment = exports.ChangeTasDomainLicenseTypeDetails = exports.ChangeTagNamespaceCompartmentDetail = exports.ChangeDomainLicenseTypeDetails = exports.ChangeDomainCompartmentDetails = exports.BulkMoveResourcesDetails = exports.BulkEditTagsResourceTypeCollection = exports.BulkEditTagsResourceType = exports.BulkEditTagsDetails = exports.BulkEditResource = exports.BulkEditOperationDetails = exports.BulkDeleteTagsDetails = exports.BulkDeleteResourcesDetails = exports.BulkActionResourceTypeCollection = exports.BulkActionResourceType = exports.BulkActionResource = exports.BaseTagDefinitionValidator = exports.AvailabilityDomain = exports.AuthenticationPolicy = exports.AuthToken = exports.ApiKey = exports.AllowedDomainLicenseTypeSummary = exports.AddUserToGroupDetails = void 0;
exports.UIPasswordInformation = exports.UIPassword = exports.Tenancy = exports.TaggingWorkRequestSummary = exports.TaggingWorkRequestLogSummary = exports.TaggingWorkRequestErrorSummary = exports.TaggingWorkRequest = exports.TagSummary = exports.TagNamespaceSummary = exports.TagNamespace = exports.TagDefaultSummary = exports.TagDefault = exports.Tag = exports.SwiftPassword = exports.StandardTagNamespaceTemplateSummary = exports.StandardTagNamespaceTemplate = exports.StandardTagDefinitionTemplate = exports.SmtpCredentialSummary = exports.SmtpCredential = exports.ScimClientCredentials = exports.ReplicatedRegionDetails = exports.RegionSubscription = exports.Region = exports.Policy = exports.PasswordPolicy = exports.OAuth2ClientCredentialSummary = exports.OAuth2ClientCredential = exports.NetworkSourcesVirtualSourceList = exports.NetworkSourcesSummary = exports.NetworkSources = exports.NetworkPolicy = exports.MoveCompartmentDetails = exports.MfaTotpToken = exports.MfaTotpDeviceSummary = exports.MfaTotpDevice = exports.ImportStandardTagsDetails = exports.IdpGroupMapping = exports.IdentityProviderGroupSummary = exports.IdentityProvider = exports.IamWorkRequestSummary = exports.IamWorkRequestResource = exports.IamWorkRequestLogSummary = exports.IamWorkRequestErrorSummary = exports.IamWorkRequest = exports.Group = exports.FullyQualifiedScope = exports.FaultDomain = exports.EnableReplicationToRegionDetails = exports.DynamicGroup = exports.DomainSummary = void 0;
exports.UpdateSaml2IdentityProviderDetails = exports.Saml2IdentityProvider = exports.EnumTagDefinitionValidator = exports.DefaultTagDefinitionValidator = exports.CreateSaml2IdentityProviderDetails = exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntry = exports.WorkRequestError = exports.WorkRequest = exports.UserGroupMembership = exports.UserCapabilities = exports.User = exports.UpdateUserDetails = exports.UpdateUserCapabilitiesDetails = exports.UpdateTagNamespaceDetails = exports.UpdateTagDetails = exports.UpdateTagDefaultDetails = exports.UpdateSwiftPasswordDetails = exports.UpdateStateDetails = exports.UpdateSmtpCredentialDetails = exports.UpdatePolicyDetails = exports.UpdateOAuth2ClientCredentialDetails = exports.UpdateNetworkSourceDetails = exports.UpdateIdpGroupMappingDetails = exports.UpdateIdentityProviderDetails = exports.UpdateGroupDetails = exports.UpdateDynamicGroupDetails = exports.UpdateDomainDetails = exports.UpdateCustomerSecretKeyDetails = exports.UpdateCompartmentDetails = exports.UpdateAuthenticationPolicyDetails = exports.UpdateAuthTokenDetails = void 0;
const AddUserToGroupDetails = __importStar(require("./add-user-to-group-details"));
exports.AddUserToGroupDetails = AddUserToGroupDetails.AddUserToGroupDetails;
const AllowedDomainLicenseTypeSummary = __importStar(require("./allowed-domain-license-type-summary"));
exports.AllowedDomainLicenseTypeSummary = AllowedDomainLicenseTypeSummary.AllowedDomainLicenseTypeSummary;
const ApiKey = __importStar(require("./api-key"));
exports.ApiKey = ApiKey.ApiKey;
const AuthToken = __importStar(require("./auth-token"));
exports.AuthToken = AuthToken.AuthToken;
const AuthenticationPolicy = __importStar(require("./authentication-policy"));
exports.AuthenticationPolicy = AuthenticationPolicy.AuthenticationPolicy;
const AvailabilityDomain = __importStar(require("./availability-domain"));
exports.AvailabilityDomain = AvailabilityDomain.AvailabilityDomain;
const BaseTagDefinitionValidator = __importStar(require("./base-tag-definition-validator"));
exports.BaseTagDefinitionValidator = BaseTagDefinitionValidator.BaseTagDefinitionValidator;
const BulkActionResource = __importStar(require("./bulk-action-resource"));
exports.BulkActionResource = BulkActionResource.BulkActionResource;
const BulkActionResourceType = __importStar(require("./bulk-action-resource-type"));
exports.BulkActionResourceType = BulkActionResourceType.BulkActionResourceType;
const BulkActionResourceTypeCollection = __importStar(require("./bulk-action-resource-type-collection"));
exports.BulkActionResourceTypeCollection = BulkActionResourceTypeCollection.BulkActionResourceTypeCollection;
const BulkDeleteResourcesDetails = __importStar(require("./bulk-delete-resources-details"));
exports.BulkDeleteResourcesDetails = BulkDeleteResourcesDetails.BulkDeleteResourcesDetails;
const BulkDeleteTagsDetails = __importStar(require("./bulk-delete-tags-details"));
exports.BulkDeleteTagsDetails = BulkDeleteTagsDetails.BulkDeleteTagsDetails;
const BulkEditOperationDetails = __importStar(require("./bulk-edit-operation-details"));
exports.BulkEditOperationDetails = BulkEditOperationDetails.BulkEditOperationDetails;
const BulkEditResource = __importStar(require("./bulk-edit-resource"));
exports.BulkEditResource = BulkEditResource.BulkEditResource;
const BulkEditTagsDetails = __importStar(require("./bulk-edit-tags-details"));
exports.BulkEditTagsDetails = BulkEditTagsDetails.BulkEditTagsDetails;
const BulkEditTagsResourceType = __importStar(require("./bulk-edit-tags-resource-type"));
exports.BulkEditTagsResourceType = BulkEditTagsResourceType.BulkEditTagsResourceType;
const BulkEditTagsResourceTypeCollection = __importStar(require("./bulk-edit-tags-resource-type-collection"));
exports.BulkEditTagsResourceTypeCollection = BulkEditTagsResourceTypeCollection.BulkEditTagsResourceTypeCollection;
const BulkMoveResourcesDetails = __importStar(require("./bulk-move-resources-details"));
exports.BulkMoveResourcesDetails = BulkMoveResourcesDetails.BulkMoveResourcesDetails;
const ChangeDomainCompartmentDetails = __importStar(require("./change-domain-compartment-details"));
exports.ChangeDomainCompartmentDetails = ChangeDomainCompartmentDetails.ChangeDomainCompartmentDetails;
const ChangeDomainLicenseTypeDetails = __importStar(require("./change-domain-license-type-details"));
exports.ChangeDomainLicenseTypeDetails = ChangeDomainLicenseTypeDetails.ChangeDomainLicenseTypeDetails;
const ChangeTagNamespaceCompartmentDetail = __importStar(require("./change-tag-namespace-compartment-detail"));
exports.ChangeTagNamespaceCompartmentDetail = ChangeTagNamespaceCompartmentDetail.ChangeTagNamespaceCompartmentDetail;
const ChangeTasDomainLicenseTypeDetails = __importStar(require("./change-tas-domain-license-type-details"));
exports.ChangeTasDomainLicenseTypeDetails = ChangeTasDomainLicenseTypeDetails.ChangeTasDomainLicenseTypeDetails;
const Compartment = __importStar(require("./compartment"));
exports.Compartment = Compartment.Compartment;
const CreateApiKeyDetails = __importStar(require("./create-api-key-details"));
exports.CreateApiKeyDetails = CreateApiKeyDetails.CreateApiKeyDetails;
const CreateAuthTokenDetails = __importStar(require("./create-auth-token-details"));
exports.CreateAuthTokenDetails = CreateAuthTokenDetails.CreateAuthTokenDetails;
const CreateCompartmentDetails = __importStar(require("./create-compartment-details"));
exports.CreateCompartmentDetails = CreateCompartmentDetails.CreateCompartmentDetails;
const CreateCustomerSecretKeyDetails = __importStar(require("./create-customer-secret-key-details"));
exports.CreateCustomerSecretKeyDetails = CreateCustomerSecretKeyDetails.CreateCustomerSecretKeyDetails;
const CreateDbCredentialDetails = __importStar(require("./create-db-credential-details"));
exports.CreateDbCredentialDetails = CreateDbCredentialDetails.CreateDbCredentialDetails;
const CreateDomainDetails = __importStar(require("./create-domain-details"));
exports.CreateDomainDetails = CreateDomainDetails.CreateDomainDetails;
const CreateDynamicGroupDetails = __importStar(require("./create-dynamic-group-details"));
exports.CreateDynamicGroupDetails = CreateDynamicGroupDetails.CreateDynamicGroupDetails;
const CreateGroupDetails = __importStar(require("./create-group-details"));
exports.CreateGroupDetails = CreateGroupDetails.CreateGroupDetails;
const CreateIdentityProviderDetails = __importStar(require("./create-identity-provider-details"));
exports.CreateIdentityProviderDetails = CreateIdentityProviderDetails.CreateIdentityProviderDetails;
const CreateIdpGroupMappingDetails = __importStar(require("./create-idp-group-mapping-details"));
exports.CreateIdpGroupMappingDetails = CreateIdpGroupMappingDetails.CreateIdpGroupMappingDetails;
const CreateNetworkSourceDetails = __importStar(require("./create-network-source-details"));
exports.CreateNetworkSourceDetails = CreateNetworkSourceDetails.CreateNetworkSourceDetails;
const CreateOAuth2ClientCredentialDetails = __importStar(require("./create-oauth2-client-credential-details"));
exports.CreateOAuth2ClientCredentialDetails = CreateOAuth2ClientCredentialDetails.CreateOAuth2ClientCredentialDetails;
const CreatePolicyDetails = __importStar(require("./create-policy-details"));
exports.CreatePolicyDetails = CreatePolicyDetails.CreatePolicyDetails;
const CreateRegionSubscriptionDetails = __importStar(require("./create-region-subscription-details"));
exports.CreateRegionSubscriptionDetails = CreateRegionSubscriptionDetails.CreateRegionSubscriptionDetails;
const CreateSmtpCredentialDetails = __importStar(require("./create-smtp-credential-details"));
exports.CreateSmtpCredentialDetails = CreateSmtpCredentialDetails.CreateSmtpCredentialDetails;
const CreateSwiftPasswordDetails = __importStar(require("./create-swift-password-details"));
exports.CreateSwiftPasswordDetails = CreateSwiftPasswordDetails.CreateSwiftPasswordDetails;
const CreateTagDefaultDetails = __importStar(require("./create-tag-default-details"));
exports.CreateTagDefaultDetails = CreateTagDefaultDetails.CreateTagDefaultDetails;
const CreateTagDetails = __importStar(require("./create-tag-details"));
exports.CreateTagDetails = CreateTagDetails.CreateTagDetails;
const CreateTagNamespaceDetails = __importStar(require("./create-tag-namespace-details"));
exports.CreateTagNamespaceDetails = CreateTagNamespaceDetails.CreateTagNamespaceDetails;
const CreateUserDetails = __importStar(require("./create-user-details"));
exports.CreateUserDetails = CreateUserDetails.CreateUserDetails;
const CustomerSecretKey = __importStar(require("./customer-secret-key"));
exports.CustomerSecretKey = CustomerSecretKey.CustomerSecretKey;
const CustomerSecretKeySummary = __importStar(require("./customer-secret-key-summary"));
exports.CustomerSecretKeySummary = CustomerSecretKeySummary.CustomerSecretKeySummary;
const DbCredential = __importStar(require("./db-credential"));
exports.DbCredential = DbCredential.DbCredential;
const DbCredentialSummary = __importStar(require("./db-credential-summary"));
exports.DbCredentialSummary = DbCredentialSummary.DbCredentialSummary;
const Domain = __importStar(require("./domain"));
exports.Domain = Domain.Domain;
const DomainReplication = __importStar(require("./domain-replication"));
exports.DomainReplication = DomainReplication.DomainReplication;
const DomainReplicationStates = __importStar(require("./domain-replication-states"));
exports.DomainReplicationStates = DomainReplicationStates.DomainReplicationStates;
const DomainSummary = __importStar(require("./domain-summary"));
exports.DomainSummary = DomainSummary.DomainSummary;
const DynamicGroup = __importStar(require("./dynamic-group"));
exports.DynamicGroup = DynamicGroup.DynamicGroup;
const EnableReplicationToRegionDetails = __importStar(require("./enable-replication-to-region-details"));
exports.EnableReplicationToRegionDetails = EnableReplicationToRegionDetails.EnableReplicationToRegionDetails;
const FaultDomain = __importStar(require("./fault-domain"));
exports.FaultDomain = FaultDomain.FaultDomain;
const FullyQualifiedScope = __importStar(require("./fully-qualified-scope"));
exports.FullyQualifiedScope = FullyQualifiedScope.FullyQualifiedScope;
const Group = __importStar(require("./group"));
exports.Group = Group.Group;
const IamWorkRequest = __importStar(require("./iam-work-request"));
exports.IamWorkRequest = IamWorkRequest.IamWorkRequest;
const IamWorkRequestErrorSummary = __importStar(require("./iam-work-request-error-summary"));
exports.IamWorkRequestErrorSummary = IamWorkRequestErrorSummary.IamWorkRequestErrorSummary;
const IamWorkRequestLogSummary = __importStar(require("./iam-work-request-log-summary"));
exports.IamWorkRequestLogSummary = IamWorkRequestLogSummary.IamWorkRequestLogSummary;
const IamWorkRequestResource = __importStar(require("./iam-work-request-resource"));
exports.IamWorkRequestResource = IamWorkRequestResource.IamWorkRequestResource;
const IamWorkRequestSummary = __importStar(require("./iam-work-request-summary"));
exports.IamWorkRequestSummary = IamWorkRequestSummary.IamWorkRequestSummary;
const IdentityProvider = __importStar(require("./identity-provider"));
exports.IdentityProvider = IdentityProvider.IdentityProvider;
const IdentityProviderGroupSummary = __importStar(require("./identity-provider-group-summary"));
exports.IdentityProviderGroupSummary = IdentityProviderGroupSummary.IdentityProviderGroupSummary;
const IdpGroupMapping = __importStar(require("./idp-group-mapping"));
exports.IdpGroupMapping = IdpGroupMapping.IdpGroupMapping;
const ImportStandardTagsDetails = __importStar(require("./import-standard-tags-details"));
exports.ImportStandardTagsDetails = ImportStandardTagsDetails.ImportStandardTagsDetails;
const MfaTotpDevice = __importStar(require("./mfa-totp-device"));
exports.MfaTotpDevice = MfaTotpDevice.MfaTotpDevice;
const MfaTotpDeviceSummary = __importStar(require("./mfa-totp-device-summary"));
exports.MfaTotpDeviceSummary = MfaTotpDeviceSummary.MfaTotpDeviceSummary;
const MfaTotpToken = __importStar(require("./mfa-totp-token"));
exports.MfaTotpToken = MfaTotpToken.MfaTotpToken;
const MoveCompartmentDetails = __importStar(require("./move-compartment-details"));
exports.MoveCompartmentDetails = MoveCompartmentDetails.MoveCompartmentDetails;
const NetworkPolicy = __importStar(require("./network-policy"));
exports.NetworkPolicy = NetworkPolicy.NetworkPolicy;
const NetworkSources = __importStar(require("./network-sources"));
exports.NetworkSources = NetworkSources.NetworkSources;
const NetworkSourcesSummary = __importStar(require("./network-sources-summary"));
exports.NetworkSourcesSummary = NetworkSourcesSummary.NetworkSourcesSummary;
const NetworkSourcesVirtualSourceList = __importStar(require("./network-sources_virtual-source-list"));
exports.NetworkSourcesVirtualSourceList = NetworkSourcesVirtualSourceList.NetworkSourcesVirtualSourceList;
const OAuth2ClientCredential = __importStar(require("./o-auth2-client-credential"));
exports.OAuth2ClientCredential = OAuth2ClientCredential.OAuth2ClientCredential;
const OAuth2ClientCredentialSummary = __importStar(require("./o-auth2-client-credential-summary"));
exports.OAuth2ClientCredentialSummary = OAuth2ClientCredentialSummary.OAuth2ClientCredentialSummary;
const PasswordPolicy = __importStar(require("./password-policy"));
exports.PasswordPolicy = PasswordPolicy.PasswordPolicy;
const Policy = __importStar(require("./policy"));
exports.Policy = Policy.Policy;
const Region = __importStar(require("./region"));
exports.Region = Region.Region;
const RegionSubscription = __importStar(require("./region-subscription"));
exports.RegionSubscription = RegionSubscription.RegionSubscription;
const ReplicatedRegionDetails = __importStar(require("./replicated-region-details"));
exports.ReplicatedRegionDetails = ReplicatedRegionDetails.ReplicatedRegionDetails;
const ScimClientCredentials = __importStar(require("./scim-client-credentials"));
exports.ScimClientCredentials = ScimClientCredentials.ScimClientCredentials;
const SmtpCredential = __importStar(require("./smtp-credential"));
exports.SmtpCredential = SmtpCredential.SmtpCredential;
const SmtpCredentialSummary = __importStar(require("./smtp-credential-summary"));
exports.SmtpCredentialSummary = SmtpCredentialSummary.SmtpCredentialSummary;
const StandardTagDefinitionTemplate = __importStar(require("./standard-tag-definition-template"));
exports.StandardTagDefinitionTemplate = StandardTagDefinitionTemplate.StandardTagDefinitionTemplate;
const StandardTagNamespaceTemplate = __importStar(require("./standard-tag-namespace-template"));
exports.StandardTagNamespaceTemplate = StandardTagNamespaceTemplate.StandardTagNamespaceTemplate;
const StandardTagNamespaceTemplateSummary = __importStar(require("./standard-tag-namespace-template-summary"));
exports.StandardTagNamespaceTemplateSummary = StandardTagNamespaceTemplateSummary.StandardTagNamespaceTemplateSummary;
const SwiftPassword = __importStar(require("./swift-password"));
exports.SwiftPassword = SwiftPassword.SwiftPassword;
const Tag = __importStar(require("./tag"));
exports.Tag = Tag.Tag;
const TagDefault = __importStar(require("./tag-default"));
exports.TagDefault = TagDefault.TagDefault;
const TagDefaultSummary = __importStar(require("./tag-default-summary"));
exports.TagDefaultSummary = TagDefaultSummary.TagDefaultSummary;
const TagNamespace = __importStar(require("./tag-namespace"));
exports.TagNamespace = TagNamespace.TagNamespace;
const TagNamespaceSummary = __importStar(require("./tag-namespace-summary"));
exports.TagNamespaceSummary = TagNamespaceSummary.TagNamespaceSummary;
const TagSummary = __importStar(require("./tag-summary"));
exports.TagSummary = TagSummary.TagSummary;
const TaggingWorkRequest = __importStar(require("./tagging-work-request"));
exports.TaggingWorkRequest = TaggingWorkRequest.TaggingWorkRequest;
const TaggingWorkRequestErrorSummary = __importStar(require("./tagging-work-request-error-summary"));
exports.TaggingWorkRequestErrorSummary = TaggingWorkRequestErrorSummary.TaggingWorkRequestErrorSummary;
const TaggingWorkRequestLogSummary = __importStar(require("./tagging-work-request-log-summary"));
exports.TaggingWorkRequestLogSummary = TaggingWorkRequestLogSummary.TaggingWorkRequestLogSummary;
const TaggingWorkRequestSummary = __importStar(require("./tagging-work-request-summary"));
exports.TaggingWorkRequestSummary = TaggingWorkRequestSummary.TaggingWorkRequestSummary;
const Tenancy = __importStar(require("./tenancy"));
exports.Tenancy = Tenancy.Tenancy;
const UIPassword = __importStar(require("./u-ipassword"));
exports.UIPassword = UIPassword.UIPassword;
const UIPasswordInformation = __importStar(require("./u-ipassword-information"));
exports.UIPasswordInformation = UIPasswordInformation.UIPasswordInformation;
const UpdateAuthTokenDetails = __importStar(require("./update-auth-token-details"));
exports.UpdateAuthTokenDetails = UpdateAuthTokenDetails.UpdateAuthTokenDetails;
const UpdateAuthenticationPolicyDetails = __importStar(require("./update-authentication-policy-details"));
exports.UpdateAuthenticationPolicyDetails = UpdateAuthenticationPolicyDetails.UpdateAuthenticationPolicyDetails;
const UpdateCompartmentDetails = __importStar(require("./update-compartment-details"));
exports.UpdateCompartmentDetails = UpdateCompartmentDetails.UpdateCompartmentDetails;
const UpdateCustomerSecretKeyDetails = __importStar(require("./update-customer-secret-key-details"));
exports.UpdateCustomerSecretKeyDetails = UpdateCustomerSecretKeyDetails.UpdateCustomerSecretKeyDetails;
const UpdateDomainDetails = __importStar(require("./update-domain-details"));
exports.UpdateDomainDetails = UpdateDomainDetails.UpdateDomainDetails;
const UpdateDynamicGroupDetails = __importStar(require("./update-dynamic-group-details"));
exports.UpdateDynamicGroupDetails = UpdateDynamicGroupDetails.UpdateDynamicGroupDetails;
const UpdateGroupDetails = __importStar(require("./update-group-details"));
exports.UpdateGroupDetails = UpdateGroupDetails.UpdateGroupDetails;
const UpdateIdentityProviderDetails = __importStar(require("./update-identity-provider-details"));
exports.UpdateIdentityProviderDetails = UpdateIdentityProviderDetails.UpdateIdentityProviderDetails;
const UpdateIdpGroupMappingDetails = __importStar(require("./update-idp-group-mapping-details"));
exports.UpdateIdpGroupMappingDetails = UpdateIdpGroupMappingDetails.UpdateIdpGroupMappingDetails;
const UpdateNetworkSourceDetails = __importStar(require("./update-network-source-details"));
exports.UpdateNetworkSourceDetails = UpdateNetworkSourceDetails.UpdateNetworkSourceDetails;
const UpdateOAuth2ClientCredentialDetails = __importStar(require("./update-oauth2-client-credential-details"));
exports.UpdateOAuth2ClientCredentialDetails = UpdateOAuth2ClientCredentialDetails.UpdateOAuth2ClientCredentialDetails;
const UpdatePolicyDetails = __importStar(require("./update-policy-details"));
exports.UpdatePolicyDetails = UpdatePolicyDetails.UpdatePolicyDetails;
const UpdateSmtpCredentialDetails = __importStar(require("./update-smtp-credential-details"));
exports.UpdateSmtpCredentialDetails = UpdateSmtpCredentialDetails.UpdateSmtpCredentialDetails;
const UpdateStateDetails = __importStar(require("./update-state-details"));
exports.UpdateStateDetails = UpdateStateDetails.UpdateStateDetails;
const UpdateSwiftPasswordDetails = __importStar(require("./update-swift-password-details"));
exports.UpdateSwiftPasswordDetails = UpdateSwiftPasswordDetails.UpdateSwiftPasswordDetails;
const UpdateTagDefaultDetails = __importStar(require("./update-tag-default-details"));
exports.UpdateTagDefaultDetails = UpdateTagDefaultDetails.UpdateTagDefaultDetails;
const UpdateTagDetails = __importStar(require("./update-tag-details"));
exports.UpdateTagDetails = UpdateTagDetails.UpdateTagDetails;
const UpdateTagNamespaceDetails = __importStar(require("./update-tag-namespace-details"));
exports.UpdateTagNamespaceDetails = UpdateTagNamespaceDetails.UpdateTagNamespaceDetails;
const UpdateUserCapabilitiesDetails = __importStar(require("./update-user-capabilities-details"));
exports.UpdateUserCapabilitiesDetails = UpdateUserCapabilitiesDetails.UpdateUserCapabilitiesDetails;
const UpdateUserDetails = __importStar(require("./update-user-details"));
exports.UpdateUserDetails = UpdateUserDetails.UpdateUserDetails;
const User = __importStar(require("./user"));
exports.User = User.User;
const UserCapabilities = __importStar(require("./user-capabilities"));
exports.UserCapabilities = UserCapabilities.UserCapabilities;
const UserGroupMembership = __importStar(require("./user-group-membership"));
exports.UserGroupMembership = UserGroupMembership.UserGroupMembership;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const CreateSaml2IdentityProviderDetails = __importStar(require("./create-saml2-identity-provider-details"));
exports.CreateSaml2IdentityProviderDetails = CreateSaml2IdentityProviderDetails.CreateSaml2IdentityProviderDetails;
const DefaultTagDefinitionValidator = __importStar(require("./default-tag-definition-validator"));
exports.DefaultTagDefinitionValidator = DefaultTagDefinitionValidator.DefaultTagDefinitionValidator;
const EnumTagDefinitionValidator = __importStar(require("./enum-tag-definition-validator"));
exports.EnumTagDefinitionValidator = EnumTagDefinitionValidator.EnumTagDefinitionValidator;
const Saml2IdentityProvider = __importStar(require("./saml2-identity-provider"));
exports.Saml2IdentityProvider = Saml2IdentityProvider.Saml2IdentityProvider;
const UpdateSaml2IdentityProviderDetails = __importStar(require("./update-saml2-identity-provider-details"));
exports.UpdateSaml2IdentityProviderDetails = UpdateSaml2IdentityProviderDetails.UpdateSaml2IdentityProviderDetails;
