// oci-identity/lib/model/replicated-region-details.js
"use strict";
/**
 * Identity and Access Management Service API
 * APIs for managing users, groups, compartments, policies, and identity domains.
 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReplicatedRegionDetails = void 0;
var ReplicatedRegionDetails;
(function (ReplicatedRegionDetails) {
    let State;
    (function (State) {
        State["EnablingReplication"] = "ENABLING_REPLICATION";
        State["ReplicationEnabled"] = "REPLICATION_ENABLED";
        State["DisablingReplication"] = "DISABLING_REPLICATION";
        State["ReplicationDisabled"] = "REPLICATION_DISABLED";
        State["Deleted"] = "DELETED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        State["UnknownValue"] = "UNKNOWN_VALUE";
    })(State = ReplicatedRegionDetails.State || (ReplicatedRegionDetails.State = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ReplicatedRegionDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ReplicatedRegionDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(ReplicatedRegionDetails = exports.ReplicatedRegionDetails || (exports.ReplicatedRegionDetails = {}));
