// oci-identity/lib/model/standard-tag-definition-template.js
"use strict";
/**
 * Identity and Access Management Service API
 * APIs for managing users, groups, compartments, policies, and identity domains.
 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StandardTagDefinitionTemplate = void 0;
var StandardTagDefinitionTemplate;
(function (StandardTagDefinitionTemplate) {
    let Type;
    (function (Type) {
        Type["Enum"] = "ENUM";
        Type["String"] = "STRING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = StandardTagDefinitionTemplate.Type || (StandardTagDefinitionTemplate.Type = {}));
    let EnumMutability;
    (function (EnumMutability) {
        EnumMutability["Immutable"] = "IMMUTABLE";
        EnumMutability["Mutable"] = "MUTABLE";
        EnumMutability["Appendable"] = "APPENDABLE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        EnumMutability["UnknownValue"] = "UNKNOWN_VALUE";
    })(EnumMutability = StandardTagDefinitionTemplate.EnumMutability || (StandardTagDefinitionTemplate.EnumMutability = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    StandardTagDefinitionTemplate.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    StandardTagDefinitionTemplate.getDeserializedJsonObj = getDeserializedJsonObj;
})(StandardTagDefinitionTemplate = exports.StandardTagDefinitionTemplate || (exports.StandardTagDefinitionTemplate = {}));
