// oci-database/lib/model/autonomous-container-database-dataguard-association.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutonomousContainerDatabaseDataguardAssociation = void 0;
var AutonomousContainerDatabaseDataguardAssociation;
(function (AutonomousContainerDatabaseDataguardAssociation) {
    let Role;
    (function (Role) {
        Role["Primary"] = "PRIMARY";
        Role["Standby"] = "STANDBY";
        Role["DisabledStandby"] = "DISABLED_STANDBY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Role["UnknownValue"] = "UNKNOWN_VALUE";
    })(Role = AutonomousContainerDatabaseDataguardAssociation.Role || (AutonomousContainerDatabaseDataguardAssociation.Role = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Provisioning"] = "PROVISIONING";
        LifecycleState["Available"] = "AVAILABLE";
        LifecycleState["RoleChangeInProgress"] = "ROLE_CHANGE_IN_PROGRESS";
        LifecycleState["Terminating"] = "TERMINATING";
        LifecycleState["Terminated"] = "TERMINATED";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["Unavailable"] = "UNAVAILABLE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = AutonomousContainerDatabaseDataguardAssociation.LifecycleState || (AutonomousContainerDatabaseDataguardAssociation.LifecycleState = {}));
    let PeerRole;
    (function (PeerRole) {
        PeerRole["Primary"] = "PRIMARY";
        PeerRole["Standby"] = "STANDBY";
        PeerRole["DisabledStandby"] = "DISABLED_STANDBY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        PeerRole["UnknownValue"] = "UNKNOWN_VALUE";
    })(PeerRole = AutonomousContainerDatabaseDataguardAssociation.PeerRole || (AutonomousContainerDatabaseDataguardAssociation.PeerRole = {}));
    let PeerLifecycleState;
    (function (PeerLifecycleState) {
        PeerLifecycleState["Provisioning"] = "PROVISIONING";
        PeerLifecycleState["Available"] = "AVAILABLE";
        PeerLifecycleState["RoleChangeInProgress"] = "ROLE_CHANGE_IN_PROGRESS";
        PeerLifecycleState["Terminating"] = "TERMINATING";
        PeerLifecycleState["Terminated"] = "TERMINATED";
        PeerLifecycleState["Failed"] = "FAILED";
        PeerLifecycleState["Unavailable"] = "UNAVAILABLE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        PeerLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(PeerLifecycleState = AutonomousContainerDatabaseDataguardAssociation.PeerLifecycleState || (AutonomousContainerDatabaseDataguardAssociation.PeerLifecycleState = {}));
    let ProtectionMode;
    (function (ProtectionMode) {
        ProtectionMode["MaximumAvailability"] = "MAXIMUM_AVAILABILITY";
        ProtectionMode["MaximumPerformance"] = "MAXIMUM_PERFORMANCE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ProtectionMode["UnknownValue"] = "UNKNOWN_VALUE";
    })(ProtectionMode = AutonomousContainerDatabaseDataguardAssociation.ProtectionMode || (AutonomousContainerDatabaseDataguardAssociation.ProtectionMode = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AutonomousContainerDatabaseDataguardAssociation.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AutonomousContainerDatabaseDataguardAssociation.getDeserializedJsonObj = getDeserializedJsonObj;
})(AutonomousContainerDatabaseDataguardAssociation = exports.AutonomousContainerDatabaseDataguardAssociation || (exports.AutonomousContainerDatabaseDataguardAssociation = {}));
