// oci-database/lib/model/autonomous-container-database-summary.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutonomousContainerDatabaseSummary = void 0;
const model = __importStar(require("../model"));
var AutonomousContainerDatabaseSummary;
(function (AutonomousContainerDatabaseSummary) {
    let ServiceLevelAgreementType;
    (function (ServiceLevelAgreementType) {
        ServiceLevelAgreementType["Standard"] = "STANDARD";
        ServiceLevelAgreementType["MissionCritical"] = "MISSION_CRITICAL";
        ServiceLevelAgreementType["AutonomousDataguard"] = "AUTONOMOUS_DATAGUARD";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ServiceLevelAgreementType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ServiceLevelAgreementType = AutonomousContainerDatabaseSummary.ServiceLevelAgreementType || (AutonomousContainerDatabaseSummary.ServiceLevelAgreementType = {}));
    let InfrastructureType;
    (function (InfrastructureType) {
        InfrastructureType["Cloud"] = "CLOUD";
        InfrastructureType["CloudAtCustomer"] = "CLOUD_AT_CUSTOMER";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        InfrastructureType["UnknownValue"] = "UNKNOWN_VALUE";
    })(InfrastructureType = AutonomousContainerDatabaseSummary.InfrastructureType || (AutonomousContainerDatabaseSummary.InfrastructureType = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Provisioning"] = "PROVISIONING";
        LifecycleState["Available"] = "AVAILABLE";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Terminating"] = "TERMINATING";
        LifecycleState["Terminated"] = "TERMINATED";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["BackupInProgress"] = "BACKUP_IN_PROGRESS";
        LifecycleState["Restoring"] = "RESTORING";
        LifecycleState["RestoreFailed"] = "RESTORE_FAILED";
        LifecycleState["Restarting"] = "RESTARTING";
        LifecycleState["MaintenanceInProgress"] = "MAINTENANCE_IN_PROGRESS";
        LifecycleState["RoleChangeInProgress"] = "ROLE_CHANGE_IN_PROGRESS";
        LifecycleState["Unavailable"] = "UNAVAILABLE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = AutonomousContainerDatabaseSummary.LifecycleState || (AutonomousContainerDatabaseSummary.LifecycleState = {}));
    let PatchModel;
    (function (PatchModel) {
        PatchModel["ReleaseUpdates"] = "RELEASE_UPDATES";
        PatchModel["ReleaseUpdateRevisions"] = "RELEASE_UPDATE_REVISIONS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        PatchModel["UnknownValue"] = "UNKNOWN_VALUE";
    })(PatchModel = AutonomousContainerDatabaseSummary.PatchModel || (AutonomousContainerDatabaseSummary.PatchModel = {}));
    let Role;
    (function (Role) {
        Role["Primary"] = "PRIMARY";
        Role["Standby"] = "STANDBY";
        Role["DisabledStandby"] = "DISABLED_STANDBY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Role["UnknownValue"] = "UNKNOWN_VALUE";
    })(Role = AutonomousContainerDatabaseSummary.Role || (AutonomousContainerDatabaseSummary.Role = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "keyHistoryEntry": obj.keyHistoryEntry
                ? obj.keyHistoryEntry.map(item => {
                    return model.AutonomousDatabaseKeyHistoryEntry.getJsonObj(item);
                })
                : undefined,
            "maintenanceWindow": obj.maintenanceWindow
                ? model.MaintenanceWindow.getJsonObj(obj.maintenanceWindow)
                : undefined,
            "backupConfig": obj.backupConfig
                ? model.AutonomousContainerDatabaseBackupConfig.getJsonObj(obj.backupConfig)
                : undefined
        });
        return jsonObj;
    }
    AutonomousContainerDatabaseSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "keyHistoryEntry": obj.keyHistoryEntry
                ? obj.keyHistoryEntry.map(item => {
                    return model.AutonomousDatabaseKeyHistoryEntry.getDeserializedJsonObj(item);
                })
                : undefined,
            "maintenanceWindow": obj.maintenanceWindow
                ? model.MaintenanceWindow.getDeserializedJsonObj(obj.maintenanceWindow)
                : undefined,
            "backupConfig": obj.backupConfig
                ? model.AutonomousContainerDatabaseBackupConfig.getDeserializedJsonObj(obj.backupConfig)
                : undefined
        });
        return jsonObj;
    }
    AutonomousContainerDatabaseSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(AutonomousContainerDatabaseSummary = exports.AutonomousContainerDatabaseSummary || (exports.AutonomousContainerDatabaseSummary = {}));
