// oci-database/lib/model/autonomous-data-warehouse.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutonomousDataWarehouse = void 0;
const model = __importStar(require("../model"));
var AutonomousDataWarehouse;
(function (AutonomousDataWarehouse) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Provisioning"] = "PROVISIONING";
        LifecycleState["Available"] = "AVAILABLE";
        LifecycleState["Stopping"] = "STOPPING";
        LifecycleState["Stopped"] = "STOPPED";
        LifecycleState["Starting"] = "STARTING";
        LifecycleState["Terminating"] = "TERMINATING";
        LifecycleState["Terminated"] = "TERMINATED";
        LifecycleState["Unavailable"] = "UNAVAILABLE";
        LifecycleState["RestoreInProgress"] = "RESTORE_IN_PROGRESS";
        LifecycleState["BackupInProgress"] = "BACKUP_IN_PROGRESS";
        LifecycleState["ScaleInProgress"] = "SCALE_IN_PROGRESS";
        LifecycleState["AvailableNeedsAttention"] = "AVAILABLE_NEEDS_ATTENTION";
        LifecycleState["Updating"] = "UPDATING";
    })(LifecycleState = AutonomousDataWarehouse.LifecycleState || (AutonomousDataWarehouse.LifecycleState = {}));
    let LicenseModel;
    (function (LicenseModel) {
        LicenseModel["LicenseIncluded"] = "LICENSE_INCLUDED";
        LicenseModel["BringYourOwnLicense"] = "BRING_YOUR_OWN_LICENSE";
    })(LicenseModel = AutonomousDataWarehouse.LicenseModel || (AutonomousDataWarehouse.LicenseModel = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "connectionStrings": obj.connectionStrings
                ? model.AutonomousDataWarehouseConnectionStrings.getJsonObj(obj.connectionStrings)
                : undefined
        });
        return jsonObj;
    }
    AutonomousDataWarehouse.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "connectionStrings": obj.connectionStrings
                ? model.AutonomousDataWarehouseConnectionStrings.getDeserializedJsonObj(obj.connectionStrings)
                : undefined
        });
        return jsonObj;
    }
    AutonomousDataWarehouse.getDeserializedJsonObj = getDeserializedJsonObj;
})(AutonomousDataWarehouse = exports.AutonomousDataWarehouse || (exports.AutonomousDataWarehouse = {}));
