// oci-database/lib/model/backup.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Backup = void 0;
var Backup;
(function (Backup) {
    let Type;
    (function (Type) {
        Type["Incremental"] = "INCREMENTAL";
        Type["Full"] = "FULL";
        Type["VirtualFull"] = "VIRTUAL_FULL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = Backup.Type || (Backup.Type = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["Restoring"] = "RESTORING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = Backup.LifecycleState || (Backup.LifecycleState = {}));
    let DatabaseEdition;
    (function (DatabaseEdition) {
        DatabaseEdition["StandardEdition"] = "STANDARD_EDITION";
        DatabaseEdition["EnterpriseEdition"] = "ENTERPRISE_EDITION";
        DatabaseEdition["EnterpriseEditionHighPerformance"] = "ENTERPRISE_EDITION_HIGH_PERFORMANCE";
        DatabaseEdition["EnterpriseEditionExtremePerformance"] = "ENTERPRISE_EDITION_EXTREME_PERFORMANCE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DatabaseEdition["UnknownValue"] = "UNKNOWN_VALUE";
    })(DatabaseEdition = Backup.DatabaseEdition || (Backup.DatabaseEdition = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    Backup.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    Backup.getDeserializedJsonObj = getDeserializedJsonObj;
})(Backup = exports.Backup || (exports.Backup = {}));
