// oci-database/lib/model/cloud-database-management-config.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloudDatabaseManagementConfig = void 0;
var CloudDatabaseManagementConfig;
(function (CloudDatabaseManagementConfig) {
    let ManagementStatus;
    (function (ManagementStatus) {
        ManagementStatus["Enabling"] = "ENABLING";
        ManagementStatus["Enabled"] = "ENABLED";
        ManagementStatus["Disabling"] = "DISABLING";
        ManagementStatus["Disabled"] = "DISABLED";
        ManagementStatus["Updating"] = "UPDATING";
        ManagementStatus["FailedEnabling"] = "FAILED_ENABLING";
        ManagementStatus["FailedDisabling"] = "FAILED_DISABLING";
        ManagementStatus["FailedUpdating"] = "FAILED_UPDATING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ManagementStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(ManagementStatus = CloudDatabaseManagementConfig.ManagementStatus || (CloudDatabaseManagementConfig.ManagementStatus = {}));
    let ManagementType;
    (function (ManagementType) {
        ManagementType["Basic"] = "BASIC";
        ManagementType["Advanced"] = "ADVANCED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ManagementType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ManagementType = CloudDatabaseManagementConfig.ManagementType || (CloudDatabaseManagementConfig.ManagementType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CloudDatabaseManagementConfig.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CloudDatabaseManagementConfig.getDeserializedJsonObj = getDeserializedJsonObj;
})(CloudDatabaseManagementConfig = exports.CloudDatabaseManagementConfig || (exports.CloudDatabaseManagementConfig = {}));
