// oci-database/lib/model/cloud-autonomous-vm-cluster.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloudAutonomousVmCluster = void 0;
var CloudAutonomousVmCluster;
(function (CloudAutonomousVmCluster) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Provisioning"] = "PROVISIONING";
        LifecycleState["Available"] = "AVAILABLE";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Terminating"] = "TERMINATING";
        LifecycleState["Terminated"] = "TERMINATED";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["MaintenanceInProgress"] = "MAINTENANCE_IN_PROGRESS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = CloudAutonomousVmCluster.LifecycleState || (CloudAutonomousVmCluster.LifecycleState = {}));
    let LicenseModel;
    (function (LicenseModel) {
        LicenseModel["LicenseIncluded"] = "LICENSE_INCLUDED";
        LicenseModel["BringYourOwnLicense"] = "BRING_YOUR_OWN_LICENSE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LicenseModel["UnknownValue"] = "UNKNOWN_VALUE";
    })(LicenseModel = CloudAutonomousVmCluster.LicenseModel || (CloudAutonomousVmCluster.LicenseModel = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CloudAutonomousVmCluster.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CloudAutonomousVmCluster.getDeserializedJsonObj = getDeserializedJsonObj;
})(CloudAutonomousVmCluster = exports.CloudAutonomousVmCluster || (exports.CloudAutonomousVmCluster = {}));
