// oci-database/lib/model/create-external-backup-job-details.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateExternalBackupJobDetails = void 0;
var CreateExternalBackupJobDetails;
(function (CreateExternalBackupJobDetails) {
    let DatabaseMode;
    (function (DatabaseMode) {
        DatabaseMode["Si"] = "SI";
        DatabaseMode["Rac"] = "RAC";
    })(DatabaseMode = CreateExternalBackupJobDetails.DatabaseMode || (CreateExternalBackupJobDetails.DatabaseMode = {}));
    let DatabaseEdition;
    (function (DatabaseEdition) {
        DatabaseEdition["StandardEdition"] = "STANDARD_EDITION";
        DatabaseEdition["EnterpriseEdition"] = "ENTERPRISE_EDITION";
        DatabaseEdition["EnterpriseEditionHighPerformance"] = "ENTERPRISE_EDITION_HIGH_PERFORMANCE";
        DatabaseEdition["EnterpriseEditionExtremePerformance"] = "ENTERPRISE_EDITION_EXTREME_PERFORMANCE";
    })(DatabaseEdition = CreateExternalBackupJobDetails.DatabaseEdition || (CreateExternalBackupJobDetails.DatabaseEdition = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CreateExternalBackupJobDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CreateExternalBackupJobDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(CreateExternalBackupJobDetails = exports.CreateExternalBackupJobDetails || (exports.CreateExternalBackupJobDetails = {}));
