// oci-database/lib/model/create-database-software-image-details.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateDatabaseSoftwareImageDetails = void 0;
var CreateDatabaseSoftwareImageDetails;
(function (CreateDatabaseSoftwareImageDetails) {
    let ImageShapeFamily;
    (function (ImageShapeFamily) {
        ImageShapeFamily["VmBmShape"] = "VM_BM_SHAPE";
        ImageShapeFamily["ExadataShape"] = "EXADATA_SHAPE";
        ImageShapeFamily["ExaccShape"] = "EXACC_SHAPE";
    })(ImageShapeFamily = CreateDatabaseSoftwareImageDetails.ImageShapeFamily || (CreateDatabaseSoftwareImageDetails.ImageShapeFamily = {}));
    let ImageType;
    (function (ImageType) {
        ImageType["GridImage"] = "GRID_IMAGE";
        ImageType["DatabaseImage"] = "DATABASE_IMAGE";
    })(ImageType = CreateDatabaseSoftwareImageDetails.ImageType || (CreateDatabaseSoftwareImageDetails.ImageType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CreateDatabaseSoftwareImageDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CreateDatabaseSoftwareImageDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(CreateDatabaseSoftwareImageDetails = exports.CreateDatabaseSoftwareImageDetails || (exports.CreateDatabaseSoftwareImageDetails = {}));
