// oci-database/lib/model/database-upgrade-history-entry-summary.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabaseUpgradeHistoryEntrySummary = void 0;
var DatabaseUpgradeHistoryEntrySummary;
(function (DatabaseUpgradeHistoryEntrySummary) {
    let Action;
    (function (Action) {
        Action["Precheck"] = "PRECHECK";
        Action["Upgrade"] = "UPGRADE";
        Action["Rollback"] = "ROLLBACK";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Action["UnknownValue"] = "UNKNOWN_VALUE";
    })(Action = DatabaseUpgradeHistoryEntrySummary.Action || (DatabaseUpgradeHistoryEntrySummary.Action = {}));
    let Source;
    (function (Source) {
        Source["DbHome"] = "DB_HOME";
        Source["DbVersion"] = "DB_VERSION";
        Source["DbSoftwareImage"] = "DB_SOFTWARE_IMAGE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Source["UnknownValue"] = "UNKNOWN_VALUE";
    })(Source = DatabaseUpgradeHistoryEntrySummary.Source || (DatabaseUpgradeHistoryEntrySummary.Source = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Succeeded"] = "SUCCEEDED";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["InProgress"] = "IN_PROGRESS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = DatabaseUpgradeHistoryEntrySummary.LifecycleState || (DatabaseUpgradeHistoryEntrySummary.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DatabaseUpgradeHistoryEntrySummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DatabaseUpgradeHistoryEntrySummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(DatabaseUpgradeHistoryEntrySummary = exports.DatabaseUpgradeHistoryEntrySummary || (exports.DatabaseUpgradeHistoryEntrySummary = {}));
