// oci-database/lib/model/database-management-config.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabaseManagementConfig = void 0;
var DatabaseManagementConfig;
(function (DatabaseManagementConfig) {
    let DatabaseManagementStatus;
    (function (DatabaseManagementStatus) {
        DatabaseManagementStatus["Enabling"] = "ENABLING";
        DatabaseManagementStatus["Enabled"] = "ENABLED";
        DatabaseManagementStatus["Disabling"] = "DISABLING";
        DatabaseManagementStatus["NotEnabled"] = "NOT_ENABLED";
        DatabaseManagementStatus["FailedEnabling"] = "FAILED_ENABLING";
        DatabaseManagementStatus["FailedDisabling"] = "FAILED_DISABLING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DatabaseManagementStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(DatabaseManagementStatus = DatabaseManagementConfig.DatabaseManagementStatus || (DatabaseManagementConfig.DatabaseManagementStatus = {}));
    let LicenseModel;
    (function (LicenseModel) {
        LicenseModel["LicenseIncluded"] = "LICENSE_INCLUDED";
        LicenseModel["BringYourOwnLicense"] = "BRING_YOUR_OWN_LICENSE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LicenseModel["UnknownValue"] = "UNKNOWN_VALUE";
    })(LicenseModel = DatabaseManagementConfig.LicenseModel || (DatabaseManagementConfig.LicenseModel = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DatabaseManagementConfig.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DatabaseManagementConfig.getDeserializedJsonObj = getDeserializedJsonObj;
})(DatabaseManagementConfig = exports.DatabaseManagementConfig || (exports.DatabaseManagementConfig = {}));
