// oci-database/lib/model/database.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Database = void 0;
const model = __importStar(require("../model"));
var Database;
(function (Database) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Provisioning"] = "PROVISIONING";
        LifecycleState["Available"] = "AVAILABLE";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["BackupInProgress"] = "BACKUP_IN_PROGRESS";
        LifecycleState["Upgrading"] = "UPGRADING";
        LifecycleState["Converting"] = "CONVERTING";
        LifecycleState["Terminating"] = "TERMINATING";
        LifecycleState["Terminated"] = "TERMINATED";
        LifecycleState["RestoreFailed"] = "RESTORE_FAILED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = Database.LifecycleState || (Database.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "dbBackupConfig": obj.dbBackupConfig
                ? model.DbBackupConfig.getJsonObj(obj.dbBackupConfig)
                : undefined,
            "connectionStrings": obj.connectionStrings
                ? model.DatabaseConnectionStrings.getJsonObj(obj.connectionStrings)
                : undefined,
            "databaseManagementConfig": obj.databaseManagementConfig
                ? model.CloudDatabaseManagementConfig.getJsonObj(obj.databaseManagementConfig)
                : undefined
        });
        return jsonObj;
    }
    Database.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "dbBackupConfig": obj.dbBackupConfig
                ? model.DbBackupConfig.getDeserializedJsonObj(obj.dbBackupConfig)
                : undefined,
            "connectionStrings": obj.connectionStrings
                ? model.DatabaseConnectionStrings.getDeserializedJsonObj(obj.connectionStrings)
                : undefined,
            "databaseManagementConfig": obj.databaseManagementConfig
                ? model.CloudDatabaseManagementConfig.getDeserializedJsonObj(obj.databaseManagementConfig)
                : undefined
        });
        return jsonObj;
    }
    Database.getDeserializedJsonObj = getDeserializedJsonObj;
})(Database = exports.Database || (exports.Database = {}));
