// oci-database/lib/model/database-software-image.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabaseSoftwareImage = void 0;
var DatabaseSoftwareImage;
(function (DatabaseSoftwareImage) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Provisioning"] = "PROVISIONING";
        LifecycleState["Available"] = "AVAILABLE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["Terminating"] = "TERMINATING";
        LifecycleState["Terminated"] = "TERMINATED";
        LifecycleState["Updating"] = "UPDATING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = DatabaseSoftwareImage.LifecycleState || (DatabaseSoftwareImage.LifecycleState = {}));
    let ImageType;
    (function (ImageType) {
        ImageType["GridImage"] = "GRID_IMAGE";
        ImageType["DatabaseImage"] = "DATABASE_IMAGE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ImageType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ImageType = DatabaseSoftwareImage.ImageType || (DatabaseSoftwareImage.ImageType = {}));
    let ImageShapeFamily;
    (function (ImageShapeFamily) {
        ImageShapeFamily["VmBmShape"] = "VM_BM_SHAPE";
        ImageShapeFamily["ExadataShape"] = "EXADATA_SHAPE";
        ImageShapeFamily["ExaccShape"] = "EXACC_SHAPE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ImageShapeFamily["UnknownValue"] = "UNKNOWN_VALUE";
    })(ImageShapeFamily = DatabaseSoftwareImage.ImageShapeFamily || (DatabaseSoftwareImage.ImageShapeFamily = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DatabaseSoftwareImage.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DatabaseSoftwareImage.getDeserializedJsonObj = getDeserializedJsonObj;
})(DatabaseSoftwareImage = exports.DatabaseSoftwareImage || (exports.DatabaseSoftwareImage = {}));
