// oci-database/lib/model/db-system-storage-performance-summary.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DbSystemStoragePerformanceSummary = void 0;
const model = __importStar(require("../model"));
var DbSystemStoragePerformanceSummary;
(function (DbSystemStoragePerformanceSummary) {
    let ShapeType;
    (function (ShapeType) {
        ShapeType["Amd"] = "AMD";
        ShapeType["Intel"] = "INTEL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ShapeType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ShapeType = DbSystemStoragePerformanceSummary.ShapeType || (DbSystemStoragePerformanceSummary.ShapeType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "dataStoragePerformanceList": obj.dataStoragePerformanceList
                ? obj.dataStoragePerformanceList.map(item => {
                    return model.StoragePerformanceDetails.getJsonObj(item);
                })
                : undefined,
            "recoStoragePerformanceList": obj.recoStoragePerformanceList
                ? obj.recoStoragePerformanceList.map(item => {
                    return model.StoragePerformanceDetails.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    DbSystemStoragePerformanceSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "dataStoragePerformanceList": obj.dataStoragePerformanceList
                ? obj.dataStoragePerformanceList.map(item => {
                    return model.StoragePerformanceDetails.getDeserializedJsonObj(item);
                })
                : undefined,
            "recoStoragePerformanceList": obj.recoStoragePerformanceList
                ? obj.recoStoragePerformanceList.map(item => {
                    return model.StoragePerformanceDetails.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    DbSystemStoragePerformanceSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(DbSystemStoragePerformanceSummary = exports.DbSystemStoragePerformanceSummary || (exports.DbSystemStoragePerformanceSummary = {}));
