// oci-database/lib/model/db-system-upgrade-history-entry-summary.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DbSystemUpgradeHistoryEntrySummary = void 0;
var DbSystemUpgradeHistoryEntrySummary;
(function (DbSystemUpgradeHistoryEntrySummary) {
    let Action;
    (function (Action) {
        Action["Precheck"] = "PRECHECK";
        Action["Rollback"] = "ROLLBACK";
        Action["UpdateSnapshotRetentionDays"] = "UPDATE_SNAPSHOT_RETENTION_DAYS";
        Action["Upgrade"] = "UPGRADE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Action["UnknownValue"] = "UNKNOWN_VALUE";
    })(Action = DbSystemUpgradeHistoryEntrySummary.Action || (DbSystemUpgradeHistoryEntrySummary.Action = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["InProgress"] = "IN_PROGRESS";
        LifecycleState["Succeeded"] = "SUCCEEDED";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = DbSystemUpgradeHistoryEntrySummary.LifecycleState || (DbSystemUpgradeHistoryEntrySummary.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DbSystemUpgradeHistoryEntrySummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DbSystemUpgradeHistoryEntrySummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(DbSystemUpgradeHistoryEntrySummary = exports.DbSystemUpgradeHistoryEntrySummary || (exports.DbSystemUpgradeHistoryEntrySummary = {}));
