// oci-database/lib/model/db-system-summary.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DbSystemSummary = void 0;
const model = __importStar(require("../model"));
var DbSystemSummary;
(function (DbSystemSummary) {
    let StorageVolumePerformanceMode;
    (function (StorageVolumePerformanceMode) {
        StorageVolumePerformanceMode["Balanced"] = "BALANCED";
        StorageVolumePerformanceMode["HighPerformance"] = "HIGH_PERFORMANCE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        StorageVolumePerformanceMode["UnknownValue"] = "UNKNOWN_VALUE";
    })(StorageVolumePerformanceMode = DbSystemSummary.StorageVolumePerformanceMode || (DbSystemSummary.StorageVolumePerformanceMode = {}));
    let DatabaseEdition;
    (function (DatabaseEdition) {
        DatabaseEdition["StandardEdition"] = "STANDARD_EDITION";
        DatabaseEdition["EnterpriseEdition"] = "ENTERPRISE_EDITION";
        DatabaseEdition["EnterpriseEditionHighPerformance"] = "ENTERPRISE_EDITION_HIGH_PERFORMANCE";
        DatabaseEdition["EnterpriseEditionExtremePerformance"] = "ENTERPRISE_EDITION_EXTREME_PERFORMANCE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DatabaseEdition["UnknownValue"] = "UNKNOWN_VALUE";
    })(DatabaseEdition = DbSystemSummary.DatabaseEdition || (DbSystemSummary.DatabaseEdition = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Provisioning"] = "PROVISIONING";
        LifecycleState["Available"] = "AVAILABLE";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Terminating"] = "TERMINATING";
        LifecycleState["Terminated"] = "TERMINATED";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["Migrated"] = "MIGRATED";
        LifecycleState["MaintenanceInProgress"] = "MAINTENANCE_IN_PROGRESS";
        LifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
        LifecycleState["Upgrading"] = "UPGRADING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = DbSystemSummary.LifecycleState || (DbSystemSummary.LifecycleState = {}));
    let DiskRedundancy;
    (function (DiskRedundancy) {
        DiskRedundancy["High"] = "HIGH";
        DiskRedundancy["Normal"] = "NORMAL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DiskRedundancy["UnknownValue"] = "UNKNOWN_VALUE";
    })(DiskRedundancy = DbSystemSummary.DiskRedundancy || (DbSystemSummary.DiskRedundancy = {}));
    let LicenseModel;
    (function (LicenseModel) {
        LicenseModel["LicenseIncluded"] = "LICENSE_INCLUDED";
        LicenseModel["BringYourOwnLicense"] = "BRING_YOUR_OWN_LICENSE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LicenseModel["UnknownValue"] = "UNKNOWN_VALUE";
    })(LicenseModel = DbSystemSummary.LicenseModel || (DbSystemSummary.LicenseModel = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "dbSystemOptions": obj.dbSystemOptions
                ? model.DbSystemOptions.getJsonObj(obj.dbSystemOptions)
                : undefined,
            "maintenanceWindow": obj.maintenanceWindow
                ? model.MaintenanceWindow.getJsonObj(obj.maintenanceWindow)
                : undefined
        });
        return jsonObj;
    }
    DbSystemSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "dbSystemOptions": obj.dbSystemOptions
                ? model.DbSystemOptions.getDeserializedJsonObj(obj.dbSystemOptions)
                : undefined,
            "maintenanceWindow": obj.maintenanceWindow
                ? model.MaintenanceWindow.getDeserializedJsonObj(obj.maintenanceWindow)
                : undefined
        });
        return jsonObj;
    }
    DbSystemSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(DbSystemSummary = exports.DbSystemSummary || (exports.DbSystemSummary = {}));
