// oci-database/lib/model/launch-db-system-from-database-details.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LaunchDbSystemFromDatabaseDetails = void 0;
const model = __importStar(require("../model"));
var LaunchDbSystemFromDatabaseDetails;
(function (LaunchDbSystemFromDatabaseDetails) {
    let DatabaseEdition;
    (function (DatabaseEdition) {
        DatabaseEdition["StandardEdition"] = "STANDARD_EDITION";
        DatabaseEdition["EnterpriseEdition"] = "ENTERPRISE_EDITION";
        DatabaseEdition["EnterpriseEditionHighPerformance"] = "ENTERPRISE_EDITION_HIGH_PERFORMANCE";
        DatabaseEdition["EnterpriseEditionExtremePerformance"] = "ENTERPRISE_EDITION_EXTREME_PERFORMANCE";
    })(DatabaseEdition = LaunchDbSystemFromDatabaseDetails.DatabaseEdition || (LaunchDbSystemFromDatabaseDetails.DatabaseEdition = {}));
    let DiskRedundancy;
    (function (DiskRedundancy) {
        DiskRedundancy["High"] = "HIGH";
        DiskRedundancy["Normal"] = "NORMAL";
    })(DiskRedundancy = LaunchDbSystemFromDatabaseDetails.DiskRedundancy || (LaunchDbSystemFromDatabaseDetails.DiskRedundancy = {}));
    let LicenseModel;
    (function (LicenseModel) {
        LicenseModel["LicenseIncluded"] = "LICENSE_INCLUDED";
        LicenseModel["BringYourOwnLicense"] = "BRING_YOUR_OWN_LICENSE";
    })(LicenseModel = LaunchDbSystemFromDatabaseDetails.LicenseModel || (LaunchDbSystemFromDatabaseDetails.LicenseModel = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.LaunchDbSystemBase.getJsonObj(obj))), {
            "dbHome": obj.dbHome
                ? model.CreateDbHomeFromDatabaseDetails.getJsonObj(obj.dbHome)
                : undefined
        });
        return jsonObj;
    }
    LaunchDbSystemFromDatabaseDetails.getJsonObj = getJsonObj;
    LaunchDbSystemFromDatabaseDetails.source = "DATABASE";
    function getDeserializedJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.LaunchDbSystemBase.getDeserializedJsonObj(obj))), {
            "dbHome": obj.dbHome
                ? model.CreateDbHomeFromDatabaseDetails.getDeserializedJsonObj(obj.dbHome)
                : undefined
        });
        return jsonObj;
    }
    LaunchDbSystemFromDatabaseDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(LaunchDbSystemFromDatabaseDetails = exports.LaunchDbSystemFromDatabaseDetails || (exports.LaunchDbSystemFromDatabaseDetails = {}));
