// oci-database/lib/model/month.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Month = void 0;
var Month;
(function (Month) {
    let Name;
    (function (Name) {
        Name["January"] = "JANUARY";
        Name["February"] = "FEBRUARY";
        Name["March"] = "MARCH";
        Name["April"] = "APRIL";
        Name["May"] = "MAY";
        Name["June"] = "JUNE";
        Name["July"] = "JULY";
        Name["August"] = "AUGUST";
        Name["September"] = "SEPTEMBER";
        Name["October"] = "OCTOBER";
        Name["November"] = "NOVEMBER";
        Name["December"] = "DECEMBER";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Name["UnknownValue"] = "UNKNOWN_VALUE";
    })(Name = Month.Name || (Month.Name = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    Month.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    Month.getDeserializedJsonObj = getDeserializedJsonObj;
})(Month = exports.Month || (exports.Month = {}));
