// oci-database/lib/model/patch-summary.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchSummary = void 0;
var PatchSummary;
(function (PatchSummary) {
    let LastAction;
    (function (LastAction) {
        LastAction["Apply"] = "APPLY";
        LastAction["Precheck"] = "PRECHECK";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LastAction["UnknownValue"] = "UNKNOWN_VALUE";
    })(LastAction = PatchSummary.LastAction || (PatchSummary.LastAction = {}));
    let AvailableActions;
    (function (AvailableActions) {
        AvailableActions["Apply"] = "APPLY";
        AvailableActions["Precheck"] = "PRECHECK";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        AvailableActions["UnknownValue"] = "UNKNOWN_VALUE";
    })(AvailableActions = PatchSummary.AvailableActions || (PatchSummary.AvailableActions = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Available"] = "AVAILABLE";
        LifecycleState["Success"] = "SUCCESS";
        LifecycleState["InProgress"] = "IN_PROGRESS";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = PatchSummary.LifecycleState || (PatchSummary.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    PatchSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    PatchSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(PatchSummary = exports.PatchSummary || (exports.PatchSummary = {}));
