// oci-database/lib/model/pdb-conversion-history-entry-summary.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PdbConversionHistoryEntrySummary = void 0;
var PdbConversionHistoryEntrySummary;
(function (PdbConversionHistoryEntrySummary) {
    let Action;
    (function (Action) {
        Action["Precheck"] = "PRECHECK";
        Action["Convert"] = "CONVERT";
        Action["Sync"] = "SYNC";
        Action["SyncRollback"] = "SYNC_ROLLBACK";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Action["UnknownValue"] = "UNKNOWN_VALUE";
    })(Action = PdbConversionHistoryEntrySummary.Action || (PdbConversionHistoryEntrySummary.Action = {}));
    let Target;
    (function (Target) {
        Target["NewDatabase"] = "NEW_DATABASE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Target["UnknownValue"] = "UNKNOWN_VALUE";
    })(Target = PdbConversionHistoryEntrySummary.Target || (PdbConversionHistoryEntrySummary.Target = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Succeeded"] = "SUCCEEDED";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["InProgress"] = "IN_PROGRESS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = PdbConversionHistoryEntrySummary.LifecycleState || (PdbConversionHistoryEntrySummary.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    PdbConversionHistoryEntrySummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    PdbConversionHistoryEntrySummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(PdbConversionHistoryEntrySummary = exports.PdbConversionHistoryEntrySummary || (exports.PdbConversionHistoryEntrySummary = {}));
