// oci-database/lib/model/stack-monitoring-config.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StackMonitoringConfig = void 0;
var StackMonitoringConfig;
(function (StackMonitoringConfig) {
    let StackMonitoringStatus;
    (function (StackMonitoringStatus) {
        StackMonitoringStatus["Enabling"] = "ENABLING";
        StackMonitoringStatus["Enabled"] = "ENABLED";
        StackMonitoringStatus["Disabling"] = "DISABLING";
        StackMonitoringStatus["NotEnabled"] = "NOT_ENABLED";
        StackMonitoringStatus["FailedEnabling"] = "FAILED_ENABLING";
        StackMonitoringStatus["FailedDisabling"] = "FAILED_DISABLING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        StackMonitoringStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(StackMonitoringStatus = StackMonitoringConfig.StackMonitoringStatus || (StackMonitoringConfig.StackMonitoringStatus = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    StackMonitoringConfig.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    StackMonitoringConfig.getDeserializedJsonObj = getDeserializedJsonObj;
})(StackMonitoringConfig = exports.StackMonitoringConfig || (exports.StackMonitoringConfig = {}));
