// oci-database/lib/model/update-autonomous-database-details.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateAutonomousDatabaseDetails = void 0;
const model = __importStar(require("../model"));
var UpdateAutonomousDatabaseDetails;
(function (UpdateAutonomousDatabaseDetails) {
    let DbWorkload;
    (function (DbWorkload) {
        DbWorkload["Oltp"] = "OLTP";
        DbWorkload["Dw"] = "DW";
        DbWorkload["Ajd"] = "AJD";
        DbWorkload["Apex"] = "APEX";
    })(DbWorkload = UpdateAutonomousDatabaseDetails.DbWorkload || (UpdateAutonomousDatabaseDetails.DbWorkload = {}));
    let LicenseModel;
    (function (LicenseModel) {
        LicenseModel["LicenseIncluded"] = "LICENSE_INCLUDED";
        LicenseModel["BringYourOwnLicense"] = "BRING_YOUR_OWN_LICENSE";
    })(LicenseModel = UpdateAutonomousDatabaseDetails.LicenseModel || (UpdateAutonomousDatabaseDetails.LicenseModel = {}));
    let RefreshableMode;
    (function (RefreshableMode) {
        RefreshableMode["Automatic"] = "AUTOMATIC";
        RefreshableMode["Manual"] = "MANUAL";
    })(RefreshableMode = UpdateAutonomousDatabaseDetails.RefreshableMode || (UpdateAutonomousDatabaseDetails.RefreshableMode = {}));
    let OpenMode;
    (function (OpenMode) {
        OpenMode["ReadOnly"] = "READ_ONLY";
        OpenMode["ReadWrite"] = "READ_WRITE";
    })(OpenMode = UpdateAutonomousDatabaseDetails.OpenMode || (UpdateAutonomousDatabaseDetails.OpenMode = {}));
    let PermissionLevel;
    (function (PermissionLevel) {
        PermissionLevel["Restricted"] = "RESTRICTED";
        PermissionLevel["Unrestricted"] = "UNRESTRICTED";
    })(PermissionLevel = UpdateAutonomousDatabaseDetails.PermissionLevel || (UpdateAutonomousDatabaseDetails.PermissionLevel = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "customerContacts": obj.customerContacts
                ? obj.customerContacts.map(item => {
                    return model.CustomerContact.getJsonObj(item);
                })
                : undefined,
            "scheduledOperations": obj.scheduledOperations
                ? obj.scheduledOperations.map(item => {
                    return model.ScheduledOperationDetails.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    UpdateAutonomousDatabaseDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "customerContacts": obj.customerContacts
                ? obj.customerContacts.map(item => {
                    return model.CustomerContact.getDeserializedJsonObj(item);
                })
                : undefined,
            "scheduledOperations": obj.scheduledOperations
                ? obj.scheduledOperations.map(item => {
                    return model.ScheduledOperationDetails.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    UpdateAutonomousDatabaseDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(UpdateAutonomousDatabaseDetails = exports.UpdateAutonomousDatabaseDetails || (exports.UpdateAutonomousDatabaseDetails = {}));
