// oci-database/lib/model/update-data-guard-association-details.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateDataGuardAssociationDetails = void 0;
var UpdateDataGuardAssociationDetails;
(function (UpdateDataGuardAssociationDetails) {
    let ProtectionMode;
    (function (ProtectionMode) {
        ProtectionMode["MaximumAvailability"] = "MAXIMUM_AVAILABILITY";
        ProtectionMode["MaximumPerformance"] = "MAXIMUM_PERFORMANCE";
        ProtectionMode["MaximumProtection"] = "MAXIMUM_PROTECTION";
    })(ProtectionMode = UpdateDataGuardAssociationDetails.ProtectionMode || (UpdateDataGuardAssociationDetails.ProtectionMode = {}));
    let TransportType;
    (function (TransportType) {
        TransportType["Sync"] = "SYNC";
        TransportType["Async"] = "ASYNC";
        TransportType["Fastsync"] = "FASTSYNC";
    })(TransportType = UpdateDataGuardAssociationDetails.TransportType || (UpdateDataGuardAssociationDetails.TransportType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UpdateDataGuardAssociationDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UpdateDataGuardAssociationDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(UpdateDataGuardAssociationDetails = exports.UpdateDataGuardAssociationDetails || (exports.UpdateDataGuardAssociationDetails = {}));
