// oci-database/lib/model/vm-cluster-update.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VmClusterUpdate = void 0;
var VmClusterUpdate;
(function (VmClusterUpdate) {
    let LastAction;
    (function (LastAction) {
        LastAction["RollingApply"] = "ROLLING_APPLY";
        LastAction["Precheck"] = "PRECHECK";
        LastAction["Rollback"] = "ROLLBACK";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LastAction["UnknownValue"] = "UNKNOWN_VALUE";
    })(LastAction = VmClusterUpdate.LastAction || (VmClusterUpdate.LastAction = {}));
    let AvailableActions;
    (function (AvailableActions) {
        AvailableActions["RollingApply"] = "ROLLING_APPLY";
        AvailableActions["Precheck"] = "PRECHECK";
        AvailableActions["Rollback"] = "ROLLBACK";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        AvailableActions["UnknownValue"] = "UNKNOWN_VALUE";
    })(AvailableActions = VmClusterUpdate.AvailableActions || (VmClusterUpdate.AvailableActions = {}));
    let UpdateType;
    (function (UpdateType) {
        UpdateType["GiUpgrade"] = "GI_UPGRADE";
        UpdateType["GiPatch"] = "GI_PATCH";
        UpdateType["OsUpdate"] = "OS_UPDATE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        UpdateType["UnknownValue"] = "UNKNOWN_VALUE";
    })(UpdateType = VmClusterUpdate.UpdateType || (VmClusterUpdate.UpdateType = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Available"] = "AVAILABLE";
        LifecycleState["Success"] = "SUCCESS";
        LifecycleState["InProgress"] = "IN_PROGRESS";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = VmClusterUpdate.LifecycleState || (VmClusterUpdate.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    VmClusterUpdate.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    VmClusterUpdate.getDeserializedJsonObj = getDeserializedJsonObj;
})(VmClusterUpdate = exports.VmClusterUpdate || (exports.VmClusterUpdate = {}));
