// oci-identity/lib/identity-waiter.js
"use strict";
/**
 * Identity and Access Management Service API
 * APIs for managing users, groups, compartments, policies, and identity domains.
 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdentityWaiter = void 0;
const models = __importStar(require("./model"));
const oci_common_1 = require("oci-common");
class IdentityWaiter {
    constructor(client, config) {
        this.client = client;
        this.config = config;
    }
    /**
     * Waits forCompartment till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetCompartmentResponse | null (null in case of 404 response)
     */
    forCompartment(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getCompartment(request), response => targetStates.includes(response.compartment.lifecycleState), targetStates.includes(models.Compartment.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forDomain till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetDomainResponse
     */
    forDomain(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericWaiter(this.config, () => this.client.getDomain(request), response => targetStates.includes(response.domain.lifecycleState));
        });
    }
    /**
     * Waits forDynamicGroup till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetDynamicGroupResponse | null (null in case of 404 response)
     */
    forDynamicGroup(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getDynamicGroup(request), response => targetStates.includes(response.dynamicGroup.lifecycleState), targetStates.includes(models.DynamicGroup.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forGroup till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetGroupResponse | null (null in case of 404 response)
     */
    forGroup(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getGroup(request), response => targetStates.includes(response.group.lifecycleState), targetStates.includes(models.Group.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forIamWorkRequest
     *
     * @param request the request to send
     * @return response returns GetIamWorkRequestResponse
     */
    forIamWorkRequest(request) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericWaiter(this.config, () => this.client.getIamWorkRequest(request), response => (response.iamWorkRequest.timeFinished ? true : false));
        });
    }
    /**
     * Waits forIdentityProvider till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetIdentityProviderResponse | null (null in case of 404 response)
     */
    forIdentityProvider(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getIdentityProvider(request), response => targetStates.includes(response.identityProvider.lifecycleState), targetStates.includes(models.IdentityProvider.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forIdpGroupMapping till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetIdpGroupMappingResponse | null (null in case of 404 response)
     */
    forIdpGroupMapping(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getIdpGroupMapping(request), response => targetStates.includes(response.idpGroupMapping.lifecycleState), targetStates.includes(models.IdpGroupMapping.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forPolicy till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetPolicyResponse | null (null in case of 404 response)
     */
    forPolicy(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getPolicy(request), response => targetStates.includes(response.policy.lifecycleState), targetStates.includes(models.Policy.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forTag till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetTagResponse | null (null in case of 404 response)
     */
    forTag(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getTag(request), response => targetStates.includes(response.tag.lifecycleState), targetStates.includes(models.Tag.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forTagDefault till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetTagDefaultResponse
     */
    forTagDefault(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericWaiter(this.config, () => this.client.getTagDefault(request), response => targetStates.includes(response.tagDefault.lifecycleState));
        });
    }
    /**
     * Waits forTagNamespace till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetTagNamespaceResponse | null (null in case of 404 response)
     */
    forTagNamespace(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getTagNamespace(request), response => targetStates.includes(response.tagNamespace.lifecycleState), targetStates.includes(models.TagNamespace.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forTaggingWorkRequest
     *
     * @param request the request to send
     * @return response returns GetTaggingWorkRequestResponse
     */
    forTaggingWorkRequest(request) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericWaiter(this.config, () => this.client.getTaggingWorkRequest(request), response => (response.taggingWorkRequest.timeFinished ? true : false));
        });
    }
    /**
     * Waits forUser till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetUserResponse | null (null in case of 404 response)
     */
    forUser(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getUser(request), response => targetStates.includes(response.user.lifecycleState), targetStates.includes(models.User.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forUserGroupMembership till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetUserGroupMembershipResponse | null (null in case of 404 response)
     */
    forUserGroupMembership(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getUserGroupMembership(request), response => targetStates.includes(response.userGroupMembership.lifecycleState), targetStates.includes(models.UserGroupMembership.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forWorkRequest
     *
     * @param request the request to send
     * @return response returns GetWorkRequestResponse
     */
    forWorkRequest(request) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericWaiter(this.config, () => this.client.getWorkRequest(request), response => (response.workRequest.timeFinished ? true : false));
        });
    }
}
exports.IdentityWaiter = IdentityWaiter;
