// oci-identity/lib/request/index.js
"use strict";
/**
 * Identity and Access Management Service API
 * APIs for managing users, groups, compartments, policies, and identity domains.
 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListUsersRequest = exports.ListPoliciesRequest = exports.ListNetworkSourcesRequest = exports.ListMfaTotpDevicesRequest = exports.ListIdentityProvidersRequest = exports.ListIamWorkRequestLogsRequest = exports.ListIamWorkRequestErrorsRequest = exports.ListGroupsRequest = exports.ListDynamicGroupsRequest = exports.ListDomainsRequest = exports.ListDbCredentialsRequest = exports.ListCompartmentsRequest = exports.ListBulkActionResourceTypesRequest = void 0;
const ListBulkActionResourceTypesRequest = __importStar(require("./list-bulk-action-resource-types-request"));
exports.ListBulkActionResourceTypesRequest = ListBulkActionResourceTypesRequest.ListBulkActionResourceTypesRequest;
const ListCompartmentsRequest = __importStar(require("./list-compartments-request"));
exports.ListCompartmentsRequest = ListCompartmentsRequest.ListCompartmentsRequest;
const ListDbCredentialsRequest = __importStar(require("./list-db-credentials-request"));
exports.ListDbCredentialsRequest = ListDbCredentialsRequest.ListDbCredentialsRequest;
const ListDomainsRequest = __importStar(require("./list-domains-request"));
exports.ListDomainsRequest = ListDomainsRequest.ListDomainsRequest;
const ListDynamicGroupsRequest = __importStar(require("./list-dynamic-groups-request"));
exports.ListDynamicGroupsRequest = ListDynamicGroupsRequest.ListDynamicGroupsRequest;
const ListGroupsRequest = __importStar(require("./list-groups-request"));
exports.ListGroupsRequest = ListGroupsRequest.ListGroupsRequest;
const ListIamWorkRequestErrorsRequest = __importStar(require("./list-iam-work-request-errors-request"));
exports.ListIamWorkRequestErrorsRequest = ListIamWorkRequestErrorsRequest.ListIamWorkRequestErrorsRequest;
const ListIamWorkRequestLogsRequest = __importStar(require("./list-iam-work-request-logs-request"));
exports.ListIamWorkRequestLogsRequest = ListIamWorkRequestLogsRequest.ListIamWorkRequestLogsRequest;
const ListIdentityProvidersRequest = __importStar(require("./list-identity-providers-request"));
exports.ListIdentityProvidersRequest = ListIdentityProvidersRequest.ListIdentityProvidersRequest;
const ListMfaTotpDevicesRequest = __importStar(require("./list-mfa-totp-devices-request"));
exports.ListMfaTotpDevicesRequest = ListMfaTotpDevicesRequest.ListMfaTotpDevicesRequest;
const ListNetworkSourcesRequest = __importStar(require("./list-network-sources-request"));
exports.ListNetworkSourcesRequest = ListNetworkSourcesRequest.ListNetworkSourcesRequest;
const ListPoliciesRequest = __importStar(require("./list-policies-request"));
exports.ListPoliciesRequest = ListPoliciesRequest.ListPoliciesRequest;
const ListUsersRequest = __importStar(require("./list-users-request"));
exports.ListUsersRequest = ListUsersRequest.ListUsersRequest;
