// oci-core/lib/computemanagement-waiter.js
"use strict";
/**
 * Core Services API
 * Use the Core Services API to manage resources such as virtual cloud networks (VCNs),
compute instances, and block storage volumes. For more information, see the console
documentation for the [Networking](/iaas/Content/Network/Concepts/overview.htm),
[Compute](/iaas/Content/Compute/Concepts/computeoverview.htm), and
[Block Volume](/iaas/Content/Block/Concepts/overview.htm) services.

 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ComputeManagementWaiter = void 0;
const models = __importStar(require("./model"));
const oci_common_1 = require("oci-common");
const oci_workrequests_1 = require("oci-workrequests");
class ComputeManagementWaiter {
    constructor(client, workRequestClient, config) {
        this.client = client;
        this.workRequestClient = workRequestClient;
        this.config = config;
    }
    /**
     * Waits forAttachInstancePoolInstance
     *
     * @param request the request to send
     * @return response returns AttachInstancePoolInstanceResponse, GetWorkRequestResponse tuple
     */
    forAttachInstancePoolInstance(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const attachInstancePoolInstanceResponse = yield this.client.attachInstancePoolInstance(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, attachInstancePoolInstanceResponse.opcWorkRequestId);
            return {
                response: attachInstancePoolInstanceResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forCreateClusterNetwork
     *
     * @param request the request to send
     * @return response returns CreateClusterNetworkResponse, GetWorkRequestResponse tuple
     */
    forCreateClusterNetwork(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const createClusterNetworkResponse = yield this.client.createClusterNetwork(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, createClusterNetworkResponse.opcWorkRequestId);
            return { response: createClusterNetworkResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forDetachInstancePoolInstance
     *
     * @param request the request to send
     * @return response returns DetachInstancePoolInstanceResponse, GetWorkRequestResponse tuple
     */
    forDetachInstancePoolInstance(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const detachInstancePoolInstanceResponse = yield this.client.detachInstancePoolInstance(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, detachInstancePoolInstanceResponse.opcWorkRequestId);
            return {
                response: detachInstancePoolInstanceResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forClusterNetwork till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetClusterNetworkResponse | null (null in case of 404 response)
     */
    forClusterNetwork(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getClusterNetwork(request), response => targetStates.includes(response.clusterNetwork.lifecycleState), targetStates.includes(models.ClusterNetwork.LifecycleState.Terminated));
        });
    }
    /**
     * Waits forInstancePool till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetInstancePoolResponse | null (null in case of 404 response)
     */
    forInstancePool(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getInstancePool(request), response => targetStates.includes(response.instancePool.lifecycleState), targetStates.includes(models.InstancePool.LifecycleState.Terminated));
        });
    }
    /**
     * Waits forInstancePoolInstance till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetInstancePoolInstanceResponse
     */
    forInstancePoolInstance(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericWaiter(this.config, () => this.client.getInstancePoolInstance(request), response => targetStates.includes(response.instancePoolInstance.lifecycleState));
        });
    }
    /**
     * Waits forInstancePoolLoadBalancerAttachment till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetInstancePoolLoadBalancerAttachmentResponse
     */
    forInstancePoolLoadBalancerAttachment(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericWaiter(this.config, () => this.client.getInstancePoolLoadBalancerAttachment(request), response => targetStates.includes(response.instancePoolLoadBalancerAttachment.lifecycleState));
        });
    }
    /**
     * Waits forLaunchInstanceConfiguration
     *
     * @param request the request to send
     * @return response returns LaunchInstanceConfigurationResponse, GetWorkRequestResponse tuple
     */
    forLaunchInstanceConfiguration(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const launchInstanceConfigurationResponse = yield this.client.launchInstanceConfiguration(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, launchInstanceConfigurationResponse.opcWorkRequestId);
            return {
                response: launchInstanceConfigurationResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forTerminateClusterNetwork
     *
     * @param request the request to send
     * @return response returns TerminateClusterNetworkResponse, GetWorkRequestResponse tuple
     */
    forTerminateClusterNetwork(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const terminateClusterNetworkResponse = yield this.client.terminateClusterNetwork(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, terminateClusterNetworkResponse.opcWorkRequestId);
            return {
                response: terminateClusterNetworkResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
}
exports.ComputeManagementWaiter = ComputeManagementWaiter;
