// oci-core/lib/compute-waiter.js
"use strict";
/**
 * Core Services API
 * Use the Core Services API to manage resources such as virtual cloud networks (VCNs),
compute instances, and block storage volumes. For more information, see the console
documentation for the [Networking](/iaas/Content/Network/Concepts/overview.htm),
[Compute](/iaas/Content/Compute/Concepts/computeoverview.htm), and
[Block Volume](/iaas/Content/Block/Concepts/overview.htm) services.

 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ComputeWaiter = void 0;
const models = __importStar(require("./model"));
const oci_common_1 = require("oci-common");
const oci_workrequests_1 = require("oci-workrequests");
class ComputeWaiter {
    constructor(client, workRequestClient, config) {
        this.client = client;
        this.workRequestClient = workRequestClient;
        this.config = config;
    }
    /**
     * Waits forChangeComputeCapacityReservationCompartment
     *
     * @param request the request to send
     * @return response returns ChangeComputeCapacityReservationCompartmentResponse, GetWorkRequestResponse tuple
     */
    forChangeComputeCapacityReservationCompartment(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const changeComputeCapacityReservationCompartmentResponse = yield this.client.changeComputeCapacityReservationCompartment(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, changeComputeCapacityReservationCompartmentResponse.opcWorkRequestId);
            return {
                response: changeComputeCapacityReservationCompartmentResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forChangeDedicatedVmHostCompartment
     *
     * @param request the request to send
     * @return response returns ChangeDedicatedVmHostCompartmentResponse, GetWorkRequestResponse tuple
     */
    forChangeDedicatedVmHostCompartment(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const changeDedicatedVmHostCompartmentResponse = yield this.client.changeDedicatedVmHostCompartment(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, changeDedicatedVmHostCompartmentResponse.opcWorkRequestId);
            return {
                response: changeDedicatedVmHostCompartmentResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forChangeInstanceCompartment
     *
     * @param request the request to send
     * @return response returns ChangeInstanceCompartmentResponse, GetWorkRequestResponse tuple
     */
    forChangeInstanceCompartment(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const changeInstanceCompartmentResponse = yield this.client.changeInstanceCompartment(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, changeInstanceCompartmentResponse.opcWorkRequestId);
            return {
                response: changeInstanceCompartmentResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forCreateComputeCapacityReservation
     *
     * @param request the request to send
     * @return response returns CreateComputeCapacityReservationResponse, GetWorkRequestResponse tuple
     */
    forCreateComputeCapacityReservation(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const createComputeCapacityReservationResponse = yield this.client.createComputeCapacityReservation(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, createComputeCapacityReservationResponse.opcWorkRequestId);
            return {
                response: createComputeCapacityReservationResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forCreateDedicatedVmHost
     *
     * @param request the request to send
     * @return response returns CreateDedicatedVmHostResponse, GetWorkRequestResponse tuple
     */
    forCreateDedicatedVmHost(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const createDedicatedVmHostResponse = yield this.client.createDedicatedVmHost(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, createDedicatedVmHostResponse.opcWorkRequestId);
            return { response: createDedicatedVmHostResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forCreateImage
     *
     * @param request the request to send
     * @return response returns CreateImageResponse, GetWorkRequestResponse tuple
     */
    forCreateImage(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const createImageResponse = yield this.client.createImage(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, createImageResponse.opcWorkRequestId);
            return { response: createImageResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forDeleteComputeCapacityReservation
     *
     * @param request the request to send
     * @return response returns DeleteComputeCapacityReservationResponse, GetWorkRequestResponse tuple
     */
    forDeleteComputeCapacityReservation(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const deleteComputeCapacityReservationResponse = yield this.client.deleteComputeCapacityReservation(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, deleteComputeCapacityReservationResponse.opcWorkRequestId);
            return {
                response: deleteComputeCapacityReservationResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forDeleteDedicatedVmHost
     *
     * @param request the request to send
     * @return response returns DeleteDedicatedVmHostResponse, GetWorkRequestResponse tuple
     */
    forDeleteDedicatedVmHost(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const deleteDedicatedVmHostResponse = yield this.client.deleteDedicatedVmHost(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, deleteDedicatedVmHostResponse.opcWorkRequestId);
            return { response: deleteDedicatedVmHostResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forExportImage
     *
     * @param request the request to send
     * @return response returns ExportImageResponse, GetWorkRequestResponse tuple
     */
    forExportImage(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const exportImageResponse = yield this.client.exportImage(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, exportImageResponse.opcWorkRequestId);
            return { response: exportImageResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forBootVolumeAttachment till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetBootVolumeAttachmentResponse
     */
    forBootVolumeAttachment(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericWaiter(this.config, () => this.client.getBootVolumeAttachment(request), response => targetStates.includes(response.bootVolumeAttachment.lifecycleState));
        });
    }
    /**
     * Waits forComputeCapacityReservation till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetComputeCapacityReservationResponse | null (null in case of 404 response)
     */
    forComputeCapacityReservation(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getComputeCapacityReservation(request), response => targetStates.includes(response.computeCapacityReservation.lifecycleState), targetStates.includes(models.ComputeCapacityReservation.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forConsoleHistory till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetConsoleHistoryResponse
     */
    forConsoleHistory(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericWaiter(this.config, () => this.client.getConsoleHistory(request), response => targetStates.includes(response.consoleHistory.lifecycleState));
        });
    }
    /**
     * Waits forDedicatedVmHost till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetDedicatedVmHostResponse | null (null in case of 404 response)
     */
    forDedicatedVmHost(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getDedicatedVmHost(request), response => targetStates.includes(response.dedicatedVmHost.lifecycleState), targetStates.includes(models.DedicatedVmHost.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forImage till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetImageResponse | null (null in case of 404 response)
     */
    forImage(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getImage(request), response => targetStates.includes(response.image.lifecycleState), targetStates.includes(models.Image.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forInstance till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetInstanceResponse | null (null in case of 404 response)
     */
    forInstance(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getInstance(request), response => targetStates.includes(response.instance.lifecycleState), targetStates.includes(models.Instance.LifecycleState.Terminated));
        });
    }
    /**
     * Waits forInstanceConsoleConnection till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetInstanceConsoleConnectionResponse | null (null in case of 404 response)
     */
    forInstanceConsoleConnection(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getInstanceConsoleConnection(request), response => targetStates.includes(response.instanceConsoleConnection.lifecycleState), targetStates.includes(models.InstanceConsoleConnection.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forVnicAttachment till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetVnicAttachmentResponse
     */
    forVnicAttachment(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericWaiter(this.config, () => this.client.getVnicAttachment(request), response => targetStates.includes(response.vnicAttachment.lifecycleState));
        });
    }
    /**
     * Waits forVolumeAttachment till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetVolumeAttachmentResponse
     */
    forVolumeAttachment(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericWaiter(this.config, () => this.client.getVolumeAttachment(request), response => targetStates.includes(response.volumeAttachment.lifecycleState));
        });
    }
    /**
     * Waits forLaunchInstance
     *
     * @param request the request to send
     * @return response returns LaunchInstanceResponse, GetWorkRequestResponse tuple
     */
    forLaunchInstance(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const launchInstanceResponse = yield this.client.launchInstance(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, launchInstanceResponse.opcWorkRequestId);
            return { response: launchInstanceResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
    /**
     * Waits forUpdateComputeCapacityReservation
     *
     * @param request the request to send
     * @return response returns UpdateComputeCapacityReservationResponse, GetWorkRequestResponse tuple
     */
    forUpdateComputeCapacityReservation(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const updateComputeCapacityReservationResponse = yield this.client.updateComputeCapacityReservation(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, updateComputeCapacityReservationResponse.opcWorkRequestId);
            return {
                response: updateComputeCapacityReservationResponse,
                workRequestResponse: getWorkRequestResponse
            };
        });
    }
    /**
     * Waits forUpdateInstance
     *
     * @param request the request to send
     * @return response returns UpdateInstanceResponse, GetWorkRequestResponse tuple
     */
    forUpdateInstance(request) {
        return __awaiter(this, void 0, void 0, function* () {
            const updateInstanceResponse = yield this.client.updateInstance(request);
            const getWorkRequestResponse = yield oci_workrequests_1.waitForWorkRequest(this.config, this.workRequestClient, updateInstanceResponse.opcWorkRequestId);
            return { response: updateInstanceResponse, workRequestResponse: getWorkRequestResponse };
        });
    }
}
exports.ComputeWaiter = ComputeWaiter;
