// oci-core/index.js
"use strict";
/**
 * Core Services API
 * Use the Core Services API to manage resources such as virtual cloud networks (VCNs),
compute instances, and block storage volumes. For more information, see the console
documentation for the [Networking](/iaas/Content/Network/Concepts/overview.htm),
[Compute](/iaas/Content/Compute/Concepts/computeoverview.htm), and
[Block Volume](/iaas/Content/Block/Concepts/overview.htm) services.

 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualNetworkWaiter = exports.VirtualNetworkClient = exports.ComputeManagementWaiter = exports.ComputeManagementClient = exports.ComputeWaiter = exports.ComputeClient = exports.BlockstorageWaiter = exports.BlockstorageClient = exports.responses = exports.requests = exports.models = void 0;
const requests = __importStar(require("./lib/request"));
exports.requests = requests;
const models = __importStar(require("./lib/model"));
exports.models = models;
const responses = __importStar(require("./lib/response"));
exports.responses = responses;
const client = __importStar(require("./lib/client"));
const blockstorage_waiter = __importStar(require("./lib/blockstorage-waiter"));
const compute_waiter = __importStar(require("./lib/compute-waiter"));
const computemanagement_waiter = __importStar(require("./lib/computemanagement-waiter"));
const virtualnetwork_waiter = __importStar(require("./lib/virtualnetwork-waiter"));
exports.BlockstorageClient = client.BlockstorageClient;
exports.BlockstorageWaiter = blockstorage_waiter.BlockstorageWaiter;
exports.ComputeClient = client.ComputeClient;
exports.ComputeWaiter = compute_waiter.ComputeWaiter;
exports.ComputeManagementClient = client.ComputeManagementClient;
exports.ComputeManagementWaiter = computemanagement_waiter.ComputeManagementWaiter;
exports.VirtualNetworkClient = client.VirtualNetworkClient;
exports.VirtualNetworkWaiter = virtualnetwork_waiter.VirtualNetworkWaiter;
