// oci-common/lib/region.js
"use strict";
/**
 * Copyright (c) 2020, 2021 Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Region = void 0;
const realm_1 = require("./realm");
const config_file_reader_1 = require("./config-file-reader");
const fs_1 = require("fs");
const region_metadata_schema_1 = require("./region-metadata-schema");
const http_1 = require("./http");
class Region {
    constructor(regionId, realm, regionCode) {
        this._realm = realm;
        this._regionId = regionId;
        if (regionCode)
            this._regionCode = regionCode;
        Region.KNOWN_REGIONS.set(regionId, this);
    }
    get realm() {
        return this._realm;
    }
    get regionId() {
        return this._regionId;
    }
    get regionCode() {
        return this._regionCode;
    }
    /**
     * Return all known Regions in this version of the SDK, except possibly the region returned by IMDS (Instance Metadata
     * Service, only available on OCI instances), since IMDS is not automatically contacted by this method.
     *
     * To ensure that this method also returns the region provided by IMDS, call {@link Region#enableInstanceMetadata()}
     * explicitly before calling {@link Region#values()}.
     *
     */
    static values() {
        if (!Region.hasCalledForImds && !Region.hasWarnedAboutValuesWithoutInstanceMetadataService) {
            console.log("Call to Regions.values() without having contacted IMDS (Instance Metadata Service, only available on OCI instances); if you do need the region from IMDS, call Region.enableInstanceMetadata() before calling Region.values()");
            Region.hasWarnedAboutValuesWithoutInstanceMetadataService = true;
        }
        Region.registerAllRegions();
        return Array.from(this.KNOWN_REGIONS.values());
    }
    /**
     *  Register all regions and sets status
     */
    static registerAllRegions() {
        if (!Region.hasUsedConfigFile) {
            Region.addRegionsFromConfigFile();
        }
        if (!Region.hasUsedEnvVar) {
            Region.addRegionFromEnvVar();
        }
    }
    static fromRegionId(regionId) {
        /*
         * load provided region from already registered regions if it exists
         * else load provided region from region configuration file if it exists
         * else load provided region from region metadeta environment variable
         * else if instance metadeta service (IMDS) has been opted in, try loading region from IMDS
         */
        if (!regionId)
            throw Error("RegionId can not be empty or undefined");
        regionId = regionId.trim().toLocaleLowerCase("en-US");
        let foundRegion = Region.KNOWN_REGIONS.get(regionId);
        if (!foundRegion) {
            Region.addRegionsFromConfigFile();
            foundRegion = Region.KNOWN_REGIONS.get(regionId);
        }
        if (!foundRegion) {
            Region.addRegionFromEnvVar();
            foundRegion = Region.KNOWN_REGIONS.get(regionId);
        }
        if (!foundRegion && Region.hasCalledForImds) {
            Region.addRegionFromImds();
            foundRegion = Region.KNOWN_REGIONS.get(regionId);
        }
        return foundRegion;
    }
    // Adds regions from the config file
    static addRegionsFromConfigFile() {
        if (!Region.hasUsedConfigFile) {
            Region.hasUsedConfigFile = true;
            let expandedRegionConfigFilePath = config_file_reader_1.ConfigFileReader.expandUserHome(Region.REGIONS_CONFIG_FILE_PATH);
            if (config_file_reader_1.ConfigFileReader.fileExists(expandedRegionConfigFilePath)) {
                try {
                    const fileContent = fs_1.readFileSync(expandedRegionConfigFilePath, "utf8");
                    const regionMetadata = JSON.parse(fileContent);
                    if (regionMetadata && regionMetadata.length > 0 && Array.isArray(regionMetadata)) {
                        regionMetadata.map(metadata => {
                            if (region_metadata_schema_1.RegionMetadataSchema.isValidSchema(metadata)) {
                                Region.register(metadata.regionIdentifier, realm_1.Realm.register(metadata.realmKey, metadata.realmDomainComponent), metadata.regionKey);
                            }
                        });
                    }
                }
                catch (error) {
                    console.log("error reading or parsing region config file");
                }
            }
        }
    }
    // Adds region from the environment variable
    static addRegionFromEnvVar() {
        if (!Region.hasUsedEnvVar) {
            Region.hasUsedEnvVar = true;
            const envVarRegionMetadata = process.env[Region.OCI_REGION_METADATA_ENV_VAR];
            if (envVarRegionMetadata) {
                try {
                    const regionMetadata = JSON.parse(envVarRegionMetadata);
                    if (region_metadata_schema_1.RegionMetadataSchema.isValidSchema(regionMetadata)) {
                        Region.register(regionMetadata.regionIdentifier, realm_1.Realm.register(regionMetadata.realmKey, regionMetadata.realmDomainComponent), regionMetadata.regionKey);
                    }
                }
                catch (error) {
                    console.log("error reading or parsing region metadata env var config file");
                }
            }
        }
    }
    // Add region from the Instance Metadata Service
    static addRegionFromImds() {
        if (Region.imdsRegionMetadata) {
            Region.register(Region.imdsRegionMetadata.regionIdentifier, realm_1.Realm.register(Region.imdsRegionMetadata.realmKey, Region.imdsRegionMetadata.realmDomainComponent), Region.imdsRegionMetadata.regionKey);
            Region.imdsRegionMetadata = undefined;
        }
    }
    /*
     * Enable instance metadata lookup for region info
     */
    static enableInstanceMetadata() {
        return __awaiter(this, void 0, void 0, function* () {
            if (!Region.hasCalledForImds) {
                Region.hasCalledForImds = true;
                try {
                    const url = Region.IMDS_BASE_URL + "instance/regionInfo/";
                    let headers = new Headers();
                    headers.append(Region.CONTENT_TYPE_HEADER, Region.CONTENT_TYPE_HEADER_VALUE);
                    headers.append(Region.AUTHORIZATION, Region.METADATA_AUTH_HEADERS);
                    const httpClient = new http_1.FetchHttpClient(null);
                    const response = yield httpClient.send({
                        uri: url,
                        method: "GET",
                        headers: headers
                    });
                    const regionMetadata = (yield response.json());
                    if (region_metadata_schema_1.RegionMetadataSchema.isValidSchema(regionMetadata)) {
                        Region.imdsRegionMetadata = regionMetadata;
                    }
                }
                catch (error) {
                    console.log("Unable to retrieve region metadata from instance metadata service, reason :" + error);
                }
            }
        });
    }
    static register(regionId, realm, regionCode) {
        if (!regionId)
            throw Error("RegionId can not be empty or undefined");
        regionId = regionId.trim().toLocaleLowerCase("en-US");
        const region = Region.KNOWN_REGIONS.get(regionId);
        if (region) {
            if (region.realm.secondLevelDomain !== realm.secondLevelDomain) {
                throw Error(" Region " +
                    regionId +
                    " is already associated with another Realm " +
                    region.realm +
                    " It cannot be re-registered with a different realm.");
            }
            return region;
        }
        if (regionCode) {
            regionCode = regionCode.trim().toLocaleLowerCase("en-US");
        }
        return new Region(regionId, realm, regionCode);
    }
    /**
     * Function to get regionId based regionStr: regionStr can be a short code or regionId
     * if it is a shortCode then we want to return the corresponding regionId
     * @param regionStr
     * @return regionId
     */
    static getRegionIdFromShortCode(regionStr) {
        regionStr = regionStr.toLocaleLowerCase();
        let region = Region.values().find(r => r.regionCode === regionStr);
        if (region) {
            return region.regionId;
        }
        // If region short code is not found in the SDK, add regions from the regions config file
        Region.addRegionsFromConfigFile();
        region = Region.values().find(r => r.regionCode === regionStr);
        if (region) {
            return region.regionId;
        }
        // else add region from environment variable, and then check for short code
        region = Region.values().find(r => r.regionCode === regionStr);
        if (region) {
            return region.regionId;
        }
        // else add region from IMDS if it has been opted in, and then check for short code
        Region.addRegionFromImds();
        region = Region.values().find(r => r.regionCode === regionStr);
        if (region) {
            return region.regionId;
        }
        return regionStr;
    }
}
exports.Region = Region;
Region.KNOWN_REGIONS = new Map();
Region.hasCalledForImds = false;
Region.hasUsedConfigFile = false;
Region.hasUsedEnvVar = false;
Region.hasWarnedAboutValuesWithoutInstanceMetadataService = false;
Region.REGIONS_CONFIG_FILE_PATH = "~/.oci/regions-config.json";
Region.OCI_REGION_METADATA_ENV_VAR = "OCI_REGION_METADATA";
Region.IMDS_BASE_URL = "http://169.254.169.254/opc/v2/";
Region.METADATA_AUTH_HEADERS = "Bearer Oracle";
Region.AUTHORIZATION = "Authorization";
Region.CONTENT_TYPE_HEADER = "Content-Type";
Region.CONTENT_TYPE_HEADER_VALUE = "application/json";
// OC1
Region.AP_CHUNCHEON_1 = Region.register("ap-chuncheon-1", realm_1.Realm.OC1, "yny");
Region.AP_MUMBAI_1 = Region.register("ap-mumbai-1", realm_1.Realm.OC1, "bom");
Region.AP_HYDERABAD_1 = Region.register("ap-hyderabad-1", realm_1.Realm.OC1, "hyd");
Region.AP_SEOUL_1 = Region.register("ap-seoul-1", realm_1.Realm.OC1, "icn");
Region.AP_SYDNEY_1 = Region.register("ap-sydney-1", realm_1.Realm.OC1, "syd");
Region.AP_MELBOURNE_1 = Region.register("ap-melbourne-1", realm_1.Realm.OC1, "mel");
Region.AP_OSAKA_1 = Region.register("ap-osaka-1", realm_1.Realm.OC1, "kix");
Region.AP_TOKYO_1 = Region.register("ap-tokyo-1", realm_1.Realm.OC1, "nrt");
Region.CA_MONTREAL_1 = Region.register("ca-montreal-1", realm_1.Realm.OC1, "yul");
Region.CA_TORONTO_1 = Region.register("ca-toronto-1", realm_1.Realm.OC1, "yyz");
Region.EU_FRANKFURT_1 = Region.register("eu-frankfurt-1", realm_1.Realm.OC1, "fra");
Region.EU_ZURICH_1 = Region.register("eu-zurich-1", realm_1.Realm.OC1, "zrh");
Region.SA_SAOPAULO_1 = Region.register("sa-saopaulo-1", realm_1.Realm.OC1, "gru");
Region.UK_CARDIFF_1 = Region.register("uk-cardiff-1", realm_1.Realm.OC1, "cwl");
Region.UK_LONDON_1 = Region.register("uk-london-1", realm_1.Realm.OC1, "lhr");
Region.US_ASHBURN_1 = Region.register("us-ashburn-1", realm_1.Realm.OC1, "iad");
Region.US_PHOENIX_1 = Region.register("us-phoenix-1", realm_1.Realm.OC1, "phx");
Region.EU_AMSTERDAM_1 = Region.register("eu-amsterdam-1", realm_1.Realm.OC1, "ams");
Region.ME_JEDDAH_1 = Region.register("me-jeddah-1", realm_1.Realm.OC1, "jed");
Region.US_SANJOSE_1 = Region.register("us-sanjose-1", realm_1.Realm.OC1, "sjc");
Region.ME_DUBAI_1 = Region.register("me-dubai-1", realm_1.Realm.OC1, "dxb");
Region.SA_SANTIAGO_1 = Region.register("sa-santiago-1", realm_1.Realm.OC1, "scl");
Region.SA_VINHEDO_1 = Region.register("sa-vinhedo-1", realm_1.Realm.OC1, "vcp");
Region.IL_JERUSALEM_1 = Region.register("il-jerusalem-1", realm_1.Realm.OC1, "mtz");
Region.EU_MARSEILLE_1 = Region.register("eu-marseille-1", realm_1.Realm.OC1, "mrs");
Region.AP_SINGAPORE_1 = Region.register("ap-singapore-1", realm_1.Realm.OC1, "sin");
Region.ME_ABUDHABI_1 = Region.register("me-abudhabi-1", realm_1.Realm.OC1, "auh");
Region.EU_MILAN_1 = Region.register("eu-milan-1", realm_1.Realm.OC1, "lin");
Region.EU_STOCKHOLM_1 = Region.register("eu-stockholm-1", realm_1.Realm.OC1, "arn");
Region.AF_JOHANNESBURG_1 = Region.register("af-johannesburg-1", realm_1.Realm.OC1, "jnb");
// OC2
Region.US_LANGLEY_1 = Region.register("us-langley-1", realm_1.Realm.OC2, "lfi");
Region.US_LUKE_1 = Region.register("us-luke-1", realm_1.Realm.OC2, "luf");
// OC3
Region.US_GOV_ASHBURN_1 = Region.register("us-gov-ashburn-1", realm_1.Realm.OC3, "ric");
Region.US_GOV_CHICAGO_1 = Region.register("us-gov-chicago-1", realm_1.Realm.OC3, "pia");
Region.US_GOV_PHOENIX_1 = Region.register("us-gov-phoenix-1", realm_1.Realm.OC3, "tus");
// OC4
Region.UK_GOV_LONDON_1 = Region.register("uk-gov-london-1", realm_1.Realm.OC4, "ltn");
Region.UK_GOV_CARDIFF_1 = Region.register("uk-gov-cardiff-1", realm_1.Realm.OC4, "brs");
// OC8
Region.AP_CHIYODA_1 = Region.register("ap-chiyoda-1", realm_1.Realm.OC8, "nja");
Region.AP_IBARAKI_1 = Region.register("ap-ibaraki-1", realm_1.Realm.OC8, "ukb");
// OC9
Region.ME_DCC_MUSCAT_1 = Region.register("me-dcc-muscat-1", realm_1.Realm.OC9, "mct");
// OC10
Region.AP_DCC_CANBERRA_1 = Region.register("ap-dcc-canberra-1", realm_1.Realm.OC10, "wga");
