// oci-database/lib/request/list-cloud-autonomous-vm-clusters-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListCloudAutonomousVmClustersRequest = void 0;
var ListCloudAutonomousVmClustersRequest;
(function (ListCloudAutonomousVmClustersRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["Timecreated"] = "TIMECREATED";
        SortBy["Displayname"] = "DISPLAYNAME";
    })(SortBy = ListCloudAutonomousVmClustersRequest.SortBy || (ListCloudAutonomousVmClustersRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListCloudAutonomousVmClustersRequest.SortOrder || (ListCloudAutonomousVmClustersRequest.SortOrder = {}));
})(ListCloudAutonomousVmClustersRequest = exports.ListCloudAutonomousVmClustersRequest || (exports.ListCloudAutonomousVmClustersRequest = {}));
