// oci-database/lib/request/list-autonomous-database-clones-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListAutonomousDatabaseClonesRequest = void 0;
var ListAutonomousDatabaseClonesRequest;
(function (ListAutonomousDatabaseClonesRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListAutonomousDatabaseClonesRequest.SortOrder || (ListAutonomousDatabaseClonesRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["None"] = "NONE";
        SortBy["Timecreated"] = "TIMECREATED";
        SortBy["Displayname"] = "DISPLAYNAME";
    })(SortBy = ListAutonomousDatabaseClonesRequest.SortBy || (ListAutonomousDatabaseClonesRequest.SortBy = {}));
    let CloneType;
    (function (CloneType) {
        CloneType["RefreshableClone"] = "REFRESHABLE_CLONE";
    })(CloneType = ListAutonomousDatabaseClonesRequest.CloneType || (ListAutonomousDatabaseClonesRequest.CloneType = {}));
})(ListAutonomousDatabaseClonesRequest = exports.ListAutonomousDatabaseClonesRequest || (exports.ListAutonomousDatabaseClonesRequest = {}));
