// oci-core/lib/model/boot-volume.js
"use strict";
/**
 * Core Services API
 * Use the Core Services API to manage resources such as virtual cloud networks (VCNs),
compute instances, and block storage volumes. For more information, see the console
documentation for the [Networking](/iaas/Content/Network/Concepts/overview.htm),
[Compute](/iaas/Content/Compute/Concepts/computeoverview.htm), and
[Block Volume](/iaas/Content/Block/Concepts/overview.htm) services.

 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BootVolume = void 0;
const model = __importStar(require("../model"));
var BootVolume;
(function (BootVolume) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Provisioning"] = "PROVISIONING";
        LifecycleState["Restoring"] = "RESTORING";
        LifecycleState["Available"] = "AVAILABLE";
        LifecycleState["Terminating"] = "TERMINATING";
        LifecycleState["Terminated"] = "TERMINATED";
        LifecycleState["Faulty"] = "FAULTY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = BootVolume.LifecycleState || (BootVolume.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "sourceDetails": obj.sourceDetails
                ? model.BootVolumeSourceDetails.getJsonObj(obj.sourceDetails)
                : undefined,
            "bootVolumeReplicas": obj.bootVolumeReplicas
                ? obj.bootVolumeReplicas.map(item => {
                    return model.BootVolumeReplicaInfo.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    BootVolume.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "sourceDetails": obj.sourceDetails
                ? model.BootVolumeSourceDetails.getDeserializedJsonObj(obj.sourceDetails)
                : undefined,
            "bootVolumeReplicas": obj.bootVolumeReplicas
                ? obj.bootVolumeReplicas.map(item => {
                    return model.BootVolumeReplicaInfo.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    BootVolume.getDeserializedJsonObj = getDeserializedJsonObj;
})(BootVolume = exports.BootVolume || (exports.BootVolume = {}));
