// oci-core/lib/model/create-virtual-circuit-details.js
"use strict";
/**
 * Core Services API
 * Use the Core Services API to manage resources such as virtual cloud networks (VCNs),
compute instances, and block storage volumes. For more information, see the console
documentation for the [Networking](/iaas/Content/Network/Concepts/overview.htm),
[Compute](/iaas/Content/Compute/Concepts/computeoverview.htm), and
[Block Volume](/iaas/Content/Block/Concepts/overview.htm) services.

 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateVirtualCircuitDetails = void 0;
const model = __importStar(require("../model"));
var CreateVirtualCircuitDetails;
(function (CreateVirtualCircuitDetails) {
    let RoutingPolicy;
    (function (RoutingPolicy) {
        RoutingPolicy["OracleServiceNetwork"] = "ORACLE_SERVICE_NETWORK";
        RoutingPolicy["Regional"] = "REGIONAL";
        RoutingPolicy["MarketLevel"] = "MARKET_LEVEL";
        RoutingPolicy["Global"] = "GLOBAL";
    })(RoutingPolicy = CreateVirtualCircuitDetails.RoutingPolicy || (CreateVirtualCircuitDetails.RoutingPolicy = {}));
    let BgpAdminState;
    (function (BgpAdminState) {
        BgpAdminState["Enabled"] = "ENABLED";
        BgpAdminState["Disabled"] = "DISABLED";
    })(BgpAdminState = CreateVirtualCircuitDetails.BgpAdminState || (CreateVirtualCircuitDetails.BgpAdminState = {}));
    let Type;
    (function (Type) {
        Type["Public"] = "PUBLIC";
        Type["Private"] = "PRIVATE";
    })(Type = CreateVirtualCircuitDetails.Type || (CreateVirtualCircuitDetails.Type = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "crossConnectMappings": obj.crossConnectMappings
                ? obj.crossConnectMappings.map(item => {
                    return model.CrossConnectMapping.getJsonObj(item);
                })
                : undefined,
            "publicPrefixes": obj.publicPrefixes
                ? obj.publicPrefixes.map(item => {
                    return model.CreateVirtualCircuitPublicPrefixDetails.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    CreateVirtualCircuitDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "crossConnectMappings": obj.crossConnectMappings
                ? obj.crossConnectMappings.map(item => {
                    return model.CrossConnectMapping.getDeserializedJsonObj(item);
                })
                : undefined,
            "publicPrefixes": obj.publicPrefixes
                ? obj.publicPrefixes.map(item => {
                    return model.CreateVirtualCircuitPublicPrefixDetails.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    CreateVirtualCircuitDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(CreateVirtualCircuitDetails = exports.CreateVirtualCircuitDetails || (exports.CreateVirtualCircuitDetails = {}));
