// oci-core/lib/model/drg-route-rule.js
"use strict";
/**
 * Core Services API
 * Use the Core Services API to manage resources such as virtual cloud networks (VCNs),
compute instances, and block storage volumes. For more information, see the console
documentation for the [Networking](/iaas/Content/Network/Concepts/overview.htm),
[Compute](/iaas/Content/Compute/Concepts/computeoverview.htm), and
[Block Volume](/iaas/Content/Block/Concepts/overview.htm) services.

 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DrgRouteRule = void 0;
var DrgRouteRule;
(function (DrgRouteRule) {
    let DestinationType;
    (function (DestinationType) {
        DestinationType["CidrBlock"] = "CIDR_BLOCK";
        DestinationType["ServiceCidrBlock"] = "SERVICE_CIDR_BLOCK";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DestinationType["UnknownValue"] = "UNKNOWN_VALUE";
    })(DestinationType = DrgRouteRule.DestinationType || (DrgRouteRule.DestinationType = {}));
    let RouteType;
    (function (RouteType) {
        RouteType["Static"] = "STATIC";
        RouteType["Dynamic"] = "DYNAMIC";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        RouteType["UnknownValue"] = "UNKNOWN_VALUE";
    })(RouteType = DrgRouteRule.RouteType || (DrgRouteRule.RouteType = {}));
    let RouteProvenance;
    (function (RouteProvenance) {
        RouteProvenance["Static"] = "STATIC";
        RouteProvenance["Vcn"] = "VCN";
        RouteProvenance["VirtualCircuit"] = "VIRTUAL_CIRCUIT";
        RouteProvenance["IpsecTunnel"] = "IPSEC_TUNNEL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        RouteProvenance["UnknownValue"] = "UNKNOWN_VALUE";
    })(RouteProvenance = DrgRouteRule.RouteProvenance || (DrgRouteRule.RouteProvenance = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DrgRouteRule.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DrgRouteRule.getDeserializedJsonObj = getDeserializedJsonObj;
})(DrgRouteRule = exports.DrgRouteRule || (exports.DrgRouteRule = {}));
