// oci-core/lib/model/phase-one-config-details.js
"use strict";
/**
 * Core Services API
 * Use the Core Services API to manage resources such as virtual cloud networks (VCNs),
compute instances, and block storage volumes. For more information, see the console
documentation for the [Networking](/iaas/Content/Network/Concepts/overview.htm),
[Compute](/iaas/Content/Compute/Concepts/computeoverview.htm), and
[Block Volume](/iaas/Content/Block/Concepts/overview.htm) services.

 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PhaseOneConfigDetails = void 0;
var PhaseOneConfigDetails;
(function (PhaseOneConfigDetails) {
    let AuthenticationAlgorithm;
    (function (AuthenticationAlgorithm) {
        AuthenticationAlgorithm["Sha2384"] = "SHA2_384";
        AuthenticationAlgorithm["Sha2256"] = "SHA2_256";
        AuthenticationAlgorithm["Sha196"] = "SHA1_96";
    })(AuthenticationAlgorithm = PhaseOneConfigDetails.AuthenticationAlgorithm || (PhaseOneConfigDetails.AuthenticationAlgorithm = {}));
    let EncryptionAlgorithm;
    (function (EncryptionAlgorithm) {
        EncryptionAlgorithm["Aes256Cbc"] = "AES_256_CBC";
        EncryptionAlgorithm["Aes192Cbc"] = "AES_192_CBC";
        EncryptionAlgorithm["Aes128Cbc"] = "AES_128_CBC";
    })(EncryptionAlgorithm = PhaseOneConfigDetails.EncryptionAlgorithm || (PhaseOneConfigDetails.EncryptionAlgorithm = {}));
    let DiffieHelmanGroup;
    (function (DiffieHelmanGroup) {
        DiffieHelmanGroup["Group2"] = "GROUP2";
        DiffieHelmanGroup["Group5"] = "GROUP5";
        DiffieHelmanGroup["Group14"] = "GROUP14";
        DiffieHelmanGroup["Group19"] = "GROUP19";
        DiffieHelmanGroup["Group20"] = "GROUP20";
        DiffieHelmanGroup["Group24"] = "GROUP24";
    })(DiffieHelmanGroup = PhaseOneConfigDetails.DiffieHelmanGroup || (PhaseOneConfigDetails.DiffieHelmanGroup = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    PhaseOneConfigDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    PhaseOneConfigDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(PhaseOneConfigDetails = exports.PhaseOneConfigDetails || (exports.PhaseOneConfigDetails = {}));
