// oci-core/lib/model/remote-peering-connection.js
"use strict";
/**
 * Core Services API
 * Use the Core Services API to manage resources such as virtual cloud networks (VCNs),
compute instances, and block storage volumes. For more information, see the console
documentation for the [Networking](/iaas/Content/Network/Concepts/overview.htm),
[Compute](/iaas/Content/Compute/Concepts/computeoverview.htm), and
[Block Volume](/iaas/Content/Block/Concepts/overview.htm) services.

 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RemotePeeringConnection = void 0;
var RemotePeeringConnection;
(function (RemotePeeringConnection) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Available"] = "AVAILABLE";
        LifecycleState["Provisioning"] = "PROVISIONING";
        LifecycleState["Terminating"] = "TERMINATING";
        LifecycleState["Terminated"] = "TERMINATED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = RemotePeeringConnection.LifecycleState || (RemotePeeringConnection.LifecycleState = {}));
    let PeeringStatus;
    (function (PeeringStatus) {
        PeeringStatus["Invalid"] = "INVALID";
        PeeringStatus["New"] = "NEW";
        PeeringStatus["Pending"] = "PENDING";
        PeeringStatus["Peered"] = "PEERED";
        PeeringStatus["Revoked"] = "REVOKED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        PeeringStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(PeeringStatus = RemotePeeringConnection.PeeringStatus || (RemotePeeringConnection.PeeringStatus = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    RemotePeeringConnection.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    RemotePeeringConnection.getDeserializedJsonObj = getDeserializedJsonObj;
})(RemotePeeringConnection = exports.RemotePeeringConnection || (exports.RemotePeeringConnection = {}));
