// oci-core/lib/model/shape-numa-nodes-per-socket-platform-options.js
"use strict";
/**
 * Core Services API
 * Use the Core Services API to manage resources such as virtual cloud networks (VCNs),
compute instances, and block storage volumes. For more information, see the console
documentation for the [Networking](/iaas/Content/Network/Concepts/overview.htm),
[Compute](/iaas/Content/Compute/Concepts/computeoverview.htm), and
[Block Volume](/iaas/Content/Block/Concepts/overview.htm) services.

 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShapeNumaNodesPerSocketPlatformOptions = void 0;
var ShapeNumaNodesPerSocketPlatformOptions;
(function (ShapeNumaNodesPerSocketPlatformOptions) {
    let AllowedValues;
    (function (AllowedValues) {
        AllowedValues["Nps0"] = "NPS0";
        AllowedValues["Nps1"] = "NPS1";
        AllowedValues["Nps2"] = "NPS2";
        AllowedValues["Nps4"] = "NPS4";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        AllowedValues["UnknownValue"] = "UNKNOWN_VALUE";
    })(AllowedValues = ShapeNumaNodesPerSocketPlatformOptions.AllowedValues || (ShapeNumaNodesPerSocketPlatformOptions.AllowedValues = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ShapeNumaNodesPerSocketPlatformOptions.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ShapeNumaNodesPerSocketPlatformOptions.getDeserializedJsonObj = getDeserializedJsonObj;
})(ShapeNumaNodesPerSocketPlatformOptions = exports.ShapeNumaNodesPerSocketPlatformOptions || (exports.ShapeNumaNodesPerSocketPlatformOptions = {}));
