// oci-core/lib/model/shape-platform-config-options.js
"use strict";
/**
 * Core Services API
 * Use the Core Services API to manage resources such as virtual cloud networks (VCNs),
compute instances, and block storage volumes. For more information, see the console
documentation for the [Networking](/iaas/Content/Network/Concepts/overview.htm),
[Compute](/iaas/Content/Compute/Concepts/computeoverview.htm), and
[Block Volume](/iaas/Content/Block/Concepts/overview.htm) services.

 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShapePlatformConfigOptions = void 0;
const model = __importStar(require("../model"));
var ShapePlatformConfigOptions;
(function (ShapePlatformConfigOptions) {
    let Type;
    (function (Type) {
        Type["AmdMilanBm"] = "AMD_MILAN_BM";
        Type["AmdRomeBm"] = "AMD_ROME_BM";
        Type["AmdRomeBmGpu"] = "AMD_ROME_BM_GPU";
        Type["IntelIcelakeBm"] = "INTEL_ICELAKE_BM";
        Type["IntelSkylakeBm"] = "INTEL_SKYLAKE_BM";
        Type["AmdVm"] = "AMD_VM";
        Type["IntelVm"] = "INTEL_VM";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = ShapePlatformConfigOptions.Type || (ShapePlatformConfigOptions.Type = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "secureBootOptions": obj.secureBootOptions
                ? model.ShapeSecureBootOptions.getJsonObj(obj.secureBootOptions)
                : undefined,
            "measuredBootOptions": obj.measuredBootOptions
                ? model.ShapeMeasuredBootOptions.getJsonObj(obj.measuredBootOptions)
                : undefined,
            "trustedPlatformModuleOptions": obj.trustedPlatformModuleOptions
                ? model.ShapeTrustedPlatformModuleOptions.getJsonObj(obj.trustedPlatformModuleOptions)
                : undefined,
            "numaNodesPerSocketPlatformOptions": obj.numaNodesPerSocketPlatformOptions
                ? model.ShapeNumaNodesPerSocketPlatformOptions.getJsonObj(obj.numaNodesPerSocketPlatformOptions)
                : undefined,
            "symmetricMultiThreadingOptions": obj.symmetricMultiThreadingOptions
                ? model.ShapeSymmetricMultiThreadingEnabledPlatformOptions.getJsonObj(obj.symmetricMultiThreadingOptions)
                : undefined,
            "accessControlServiceOptions": obj.accessControlServiceOptions
                ? model.ShapeAccessControlServiceEnabledPlatformOptions.getJsonObj(obj.accessControlServiceOptions)
                : undefined,
            "virtualInstructionsOptions": obj.virtualInstructionsOptions
                ? model.ShapeVirtualInstructionsEnabledPlatformOptions.getJsonObj(obj.virtualInstructionsOptions)
                : undefined,
            "inputOutputMemoryManagementUnitOptions": obj.inputOutputMemoryManagementUnitOptions
                ? model.ShapeInputOutputMemoryManagementUnitEnabledPlatformOptions.getJsonObj(obj.inputOutputMemoryManagementUnitOptions)
                : undefined,
            "percentageOfCoresEnabledOptions": obj.percentageOfCoresEnabledOptions
                ? model.PercentageOfCoresEnabledOptions.getJsonObj(obj.percentageOfCoresEnabledOptions)
                : undefined
        });
        return jsonObj;
    }
    ShapePlatformConfigOptions.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "secureBootOptions": obj.secureBootOptions
                ? model.ShapeSecureBootOptions.getDeserializedJsonObj(obj.secureBootOptions)
                : undefined,
            "measuredBootOptions": obj.measuredBootOptions
                ? model.ShapeMeasuredBootOptions.getDeserializedJsonObj(obj.measuredBootOptions)
                : undefined,
            "trustedPlatformModuleOptions": obj.trustedPlatformModuleOptions
                ? model.ShapeTrustedPlatformModuleOptions.getDeserializedJsonObj(obj.trustedPlatformModuleOptions)
                : undefined,
            "numaNodesPerSocketPlatformOptions": obj.numaNodesPerSocketPlatformOptions
                ? model.ShapeNumaNodesPerSocketPlatformOptions.getDeserializedJsonObj(obj.numaNodesPerSocketPlatformOptions)
                : undefined,
            "symmetricMultiThreadingOptions": obj.symmetricMultiThreadingOptions
                ? model.ShapeSymmetricMultiThreadingEnabledPlatformOptions.getDeserializedJsonObj(obj.symmetricMultiThreadingOptions)
                : undefined,
            "accessControlServiceOptions": obj.accessControlServiceOptions
                ? model.ShapeAccessControlServiceEnabledPlatformOptions.getDeserializedJsonObj(obj.accessControlServiceOptions)
                : undefined,
            "virtualInstructionsOptions": obj.virtualInstructionsOptions
                ? model.ShapeVirtualInstructionsEnabledPlatformOptions.getDeserializedJsonObj(obj.virtualInstructionsOptions)
                : undefined,
            "inputOutputMemoryManagementUnitOptions": obj.inputOutputMemoryManagementUnitOptions
                ? model.ShapeInputOutputMemoryManagementUnitEnabledPlatformOptions.getDeserializedJsonObj(obj.inputOutputMemoryManagementUnitOptions)
                : undefined,
            "percentageOfCoresEnabledOptions": obj.percentageOfCoresEnabledOptions
                ? model.PercentageOfCoresEnabledOptions.getDeserializedJsonObj(obj.percentageOfCoresEnabledOptions)
                : undefined
        });
        return jsonObj;
    }
    ShapePlatformConfigOptions.getDeserializedJsonObj = getDeserializedJsonObj;
})(ShapePlatformConfigOptions = exports.ShapePlatformConfigOptions || (exports.ShapePlatformConfigOptions = {}));
