// oci-core/lib/model/virtual-circuit-public-prefix.js
"use strict";
/**
 * Core Services API
 * Use the Core Services API to manage resources such as virtual cloud networks (VCNs),
compute instances, and block storage volumes. For more information, see the console
documentation for the [Networking](/iaas/Content/Network/Concepts/overview.htm),
[Compute](/iaas/Content/Compute/Concepts/computeoverview.htm), and
[Block Volume](/iaas/Content/Block/Concepts/overview.htm) services.

 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualCircuitPublicPrefix = void 0;
var VirtualCircuitPublicPrefix;
(function (VirtualCircuitPublicPrefix) {
    let VerificationState;
    (function (VerificationState) {
        VerificationState["InProgress"] = "IN_PROGRESS";
        VerificationState["Completed"] = "COMPLETED";
        VerificationState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        VerificationState["UnknownValue"] = "UNKNOWN_VALUE";
    })(VerificationState = VirtualCircuitPublicPrefix.VerificationState || (VirtualCircuitPublicPrefix.VerificationState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    VirtualCircuitPublicPrefix.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    VirtualCircuitPublicPrefix.getDeserializedJsonObj = getDeserializedJsonObj;
})(VirtualCircuitPublicPrefix = exports.VirtualCircuitPublicPrefix || (exports.VirtualCircuitPublicPrefix = {}));
