// oci-core/lib/model/volume-group-backup.js
"use strict";
/**
 * Core Services API
 * Use the Core Services API to manage resources such as virtual cloud networks (VCNs),
compute instances, and block storage volumes. For more information, see the console
documentation for the [Networking](/iaas/Content/Network/Concepts/overview.htm),
[Compute](/iaas/Content/Compute/Concepts/computeoverview.htm), and
[Block Volume](/iaas/Content/Block/Concepts/overview.htm) services.

 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VolumeGroupBackup = void 0;
var VolumeGroupBackup;
(function (VolumeGroupBackup) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Committed"] = "COMMITTED";
        LifecycleState["Available"] = "AVAILABLE";
        LifecycleState["Terminating"] = "TERMINATING";
        LifecycleState["Terminated"] = "TERMINATED";
        LifecycleState["Faulty"] = "FAULTY";
        LifecycleState["RequestReceived"] = "REQUEST_RECEIVED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = VolumeGroupBackup.LifecycleState || (VolumeGroupBackup.LifecycleState = {}));
    let SourceType;
    (function (SourceType) {
        SourceType["Manual"] = "MANUAL";
        SourceType["Scheduled"] = "SCHEDULED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        SourceType["UnknownValue"] = "UNKNOWN_VALUE";
    })(SourceType = VolumeGroupBackup.SourceType || (VolumeGroupBackup.SourceType = {}));
    let Type;
    (function (Type) {
        Type["Full"] = "FULL";
        Type["Incremental"] = "INCREMENTAL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = VolumeGroupBackup.Type || (VolumeGroupBackup.Type = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    VolumeGroupBackup.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    VolumeGroupBackup.getDeserializedJsonObj = getDeserializedJsonObj;
})(VolumeGroupBackup = exports.VolumeGroupBackup || (exports.VolumeGroupBackup = {}));
