// oci-core/lib/model/volume-backup-schedule.js
"use strict";
/**
 * Core Services API
 * Use the Core Services API to manage resources such as virtual cloud networks (VCNs),
compute instances, and block storage volumes. For more information, see the console
documentation for the [Networking](/iaas/Content/Network/Concepts/overview.htm),
[Compute](/iaas/Content/Compute/Concepts/computeoverview.htm), and
[Block Volume](/iaas/Content/Block/Concepts/overview.htm) services.

 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VolumeBackupSchedule = void 0;
var VolumeBackupSchedule;
(function (VolumeBackupSchedule) {
    let BackupType;
    (function (BackupType) {
        BackupType["Full"] = "FULL";
        BackupType["Incremental"] = "INCREMENTAL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        BackupType["UnknownValue"] = "UNKNOWN_VALUE";
    })(BackupType = VolumeBackupSchedule.BackupType || (VolumeBackupSchedule.BackupType = {}));
    let Period;
    (function (Period) {
        Period["OneHour"] = "ONE_HOUR";
        Period["OneDay"] = "ONE_DAY";
        Period["OneWeek"] = "ONE_WEEK";
        Period["OneMonth"] = "ONE_MONTH";
        Period["OneYear"] = "ONE_YEAR";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Period["UnknownValue"] = "UNKNOWN_VALUE";
    })(Period = VolumeBackupSchedule.Period || (VolumeBackupSchedule.Period = {}));
    let OffsetType;
    (function (OffsetType) {
        OffsetType["Structured"] = "STRUCTURED";
        OffsetType["NumericSeconds"] = "NUMERIC_SECONDS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        OffsetType["UnknownValue"] = "UNKNOWN_VALUE";
    })(OffsetType = VolumeBackupSchedule.OffsetType || (VolumeBackupSchedule.OffsetType = {}));
    let DayOfWeek;
    (function (DayOfWeek) {
        DayOfWeek["Monday"] = "MONDAY";
        DayOfWeek["Tuesday"] = "TUESDAY";
        DayOfWeek["Wednesday"] = "WEDNESDAY";
        DayOfWeek["Thursday"] = "THURSDAY";
        DayOfWeek["Friday"] = "FRIDAY";
        DayOfWeek["Saturday"] = "SATURDAY";
        DayOfWeek["Sunday"] = "SUNDAY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DayOfWeek["UnknownValue"] = "UNKNOWN_VALUE";
    })(DayOfWeek = VolumeBackupSchedule.DayOfWeek || (VolumeBackupSchedule.DayOfWeek = {}));
    let Month;
    (function (Month) {
        Month["January"] = "JANUARY";
        Month["February"] = "FEBRUARY";
        Month["March"] = "MARCH";
        Month["April"] = "APRIL";
        Month["May"] = "MAY";
        Month["June"] = "JUNE";
        Month["July"] = "JULY";
        Month["August"] = "AUGUST";
        Month["September"] = "SEPTEMBER";
        Month["October"] = "OCTOBER";
        Month["November"] = "NOVEMBER";
        Month["December"] = "DECEMBER";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Month["UnknownValue"] = "UNKNOWN_VALUE";
    })(Month = VolumeBackupSchedule.Month || (VolumeBackupSchedule.Month = {}));
    let TimeZone;
    (function (TimeZone) {
        TimeZone["Utc"] = "UTC";
        TimeZone["RegionalDataCenterTime"] = "REGIONAL_DATA_CENTER_TIME";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        TimeZone["UnknownValue"] = "UNKNOWN_VALUE";
    })(TimeZone = VolumeBackupSchedule.TimeZone || (VolumeBackupSchedule.TimeZone = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    VolumeBackupSchedule.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    VolumeBackupSchedule.getDeserializedJsonObj = getDeserializedJsonObj;
})(VolumeBackupSchedule = exports.VolumeBackupSchedule || (exports.VolumeBackupSchedule = {}));
