// oci-core/lib/model/vtap.js
"use strict";
/**
 * Core Services API
 * Use the Core Services API to manage resources such as virtual cloud networks (VCNs),
compute instances, and block storage volumes. For more information, see the console
documentation for the [Networking](/iaas/Content/Network/Concepts/overview.htm),
[Compute](/iaas/Content/Compute/Concepts/computeoverview.htm), and
[Block Volume](/iaas/Content/Block/Concepts/overview.htm) services.

 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Vtap = void 0;
var Vtap;
(function (Vtap) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Provisioning"] = "PROVISIONING";
        LifecycleState["Available"] = "AVAILABLE";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Terminating"] = "TERMINATING";
        LifecycleState["Terminated"] = "TERMINATED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = Vtap.LifecycleState || (Vtap.LifecycleState = {}));
    let LifecycleStateDetails;
    (function (LifecycleStateDetails) {
        LifecycleStateDetails["Running"] = "RUNNING";
        LifecycleStateDetails["Stopped"] = "STOPPED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleStateDetails["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleStateDetails = Vtap.LifecycleStateDetails || (Vtap.LifecycleStateDetails = {}));
    let EncapsulationProtocol;
    (function (EncapsulationProtocol) {
        EncapsulationProtocol["Vxlan"] = "VXLAN";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        EncapsulationProtocol["UnknownValue"] = "UNKNOWN_VALUE";
    })(EncapsulationProtocol = Vtap.EncapsulationProtocol || (Vtap.EncapsulationProtocol = {}));
    let SourceType;
    (function (SourceType) {
        SourceType["Vnic"] = "VNIC";
        SourceType["Subnet"] = "SUBNET";
        SourceType["LoadBalancer"] = "LOAD_BALANCER";
        SourceType["DbSystem"] = "DB_SYSTEM";
        SourceType["ExadataVmCluster"] = "EXADATA_VM_CLUSTER";
        SourceType["AutonomousDataWarehouse"] = "AUTONOMOUS_DATA_WAREHOUSE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        SourceType["UnknownValue"] = "UNKNOWN_VALUE";
    })(SourceType = Vtap.SourceType || (Vtap.SourceType = {}));
    let TrafficMode;
    (function (TrafficMode) {
        TrafficMode["Default"] = "DEFAULT";
        TrafficMode["Priority"] = "PRIORITY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        TrafficMode["UnknownValue"] = "UNKNOWN_VALUE";
    })(TrafficMode = Vtap.TrafficMode || (Vtap.TrafficMode = {}));
    let TargetType;
    (function (TargetType) {
        TargetType["Vnic"] = "VNIC";
        TargetType["NetworkLoadBalancer"] = "NETWORK_LOAD_BALANCER";
        TargetType["IpAddress"] = "IP_ADDRESS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        TargetType["UnknownValue"] = "UNKNOWN_VALUE";
    })(TargetType = Vtap.TargetType || (Vtap.TargetType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    Vtap.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    Vtap.getDeserializedJsonObj = getDeserializedJsonObj;
})(Vtap = exports.Vtap || (exports.Vtap = {}));
