// oci-core/lib/request/list-public-ips-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListPublicIpsRequest = void 0;
var ListPublicIpsRequest;
(function (ListPublicIpsRequest) {
    let Scope;
    (function (Scope) {
        Scope["Region"] = "REGION";
        Scope["AvailabilityDomain"] = "AVAILABILITY_DOMAIN";
    })(Scope = ListPublicIpsRequest.Scope || (ListPublicIpsRequest.Scope = {}));
    let Lifetime;
    (function (Lifetime) {
        Lifetime["Ephemeral"] = "EPHEMERAL";
        Lifetime["Reserved"] = "RESERVED";
    })(Lifetime = ListPublicIpsRequest.Lifetime || (ListPublicIpsRequest.Lifetime = {}));
})(ListPublicIpsRequest = exports.ListPublicIpsRequest || (exports.ListPublicIpsRequest = {}));
