// oci-core/lib/request/list-compute-global-image-capability-schemas-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListComputeGlobalImageCapabilitySchemasRequest = void 0;
var ListComputeGlobalImageCapabilitySchemasRequest;
(function (ListComputeGlobalImageCapabilitySchemasRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["Timecreated"] = "TIMECREATED";
        SortBy["Displayname"] = "DISPLAYNAME";
    })(SortBy = ListComputeGlobalImageCapabilitySchemasRequest.SortBy || (ListComputeGlobalImageCapabilitySchemasRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListComputeGlobalImageCapabilitySchemasRequest.SortOrder || (ListComputeGlobalImageCapabilitySchemasRequest.SortOrder = {}));
})(ListComputeGlobalImageCapabilitySchemasRequest = exports.ListComputeGlobalImageCapabilitySchemasRequest || (exports.ListComputeGlobalImageCapabilitySchemasRequest = {}));
